
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A comprehensive, hands-on tutorial for ChIP-seq data analysis">
      
      
        <meta name="author" content="Raja Ishaq Nabi Khan">
      
      
        <link rel="canonical" href="https://ishaaq34.github.io/Chipseq_analysis_tutorial_mkdocs/05_alignment_bowtie2/">
      
      
        <link rel="prev" href="../04_fastq_concepts/">
      
      
        <link rel="next" href="../06_duplicate_removal_qc/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>Alignment with Bowtie2 - ChIP-seq Analysis Tutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="light-blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#alignment-solving-the-jigsaw-puzzle" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-header__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq Analysis Tutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Alignment with Bowtie2
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="light-blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="light-blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../00_introduction/" class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../01_setup_environment/" class="md-tabs__link">
          
  
  
  Setup & Data

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="./" class="md-tabs__link">
          
  
  
  Alignment & QC

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../11_macs3_peak_calling/" class="md-tabs__link">
          
  
  
  Peak Calling

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../14_bigwig_generation/" class="md-tabs__link">
          
  
  
  Visualization

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../tutorial_book_creation.md" class="md-tabs__link">
          
  
  
  Reference

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-nav__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    ChIP-seq Analysis Tutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Setup & Data
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Setup & Data
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_setup_environment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Environment Setup
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_bash_automation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Bash Automation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_geo_fastq_download/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GEO Fastq Download
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_fastq_concepts/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fastq Concepts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Alignment & QC
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Alignment & QC
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Alignment with Bowtie2
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Alignment with Bowtie2
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#basic-concept-the-puzzle" class="md-nav__link">
    <span class="md-ellipsis">
      
        Basic Concept (The "Puzzle")
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#execution-solving-it" class="md-nav__link">
    <span class="md-ellipsis">
      
        Execution (Solving It)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Execution (Solving It)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-the-index-building-the-map" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1: The Index (Building the Map)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#after-indexing-bowtie2" class="md-nav__link">
    <span class="md-ellipsis">
      
        After Indexing (Bowtie2)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-single-sample-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2: Single Sample Alignment
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-structure" class="md-nav__link">
    <span class="md-ellipsis">
      
        Output Structure
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-automation-loop" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3: Automation Loop
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optimization" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optimization
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Optimization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#optimization-threads-vs-jobs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optimization: Threads vs. Jobs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-bam-file-structure" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding BAM File Structure
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multimapping-the-lost-gps" class="md-nav__link">
    <span class="md-ellipsis">
      
        Multimapping &amp; The "Lost GPS"
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-check-samtools-flagstat-samtools-stats" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Check: samtools flagstat &amp; samtools stats
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Quality Check: samtools flagstat &amp; samtools stats">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quick-manual-mapq-check" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Manual MAPQ Check
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-multi-mapping-reads" class="md-nav__link">
    <span class="md-ellipsis">
      
        Filtering Multi-mapping Reads
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#directory-structure-after-alignment-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directory Structure After Alignment QC
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        Summary
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06_duplicate_removal_qc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Duplicate Removal & QC
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_library_complexity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Library Complexity
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../08_bam_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Quality Metrics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../09_strand_cross_correlation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Strand Cross Correlation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../10_bam_summary_fingerprint/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Summary & Fingerprint
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Peak Calling
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Peak Calling
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../11_macs3_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MACS3 Peak Calling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../12_frip_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Metrics - FRiP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../13_idr_consensus_motifs_rk_corrected/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Visualization
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Visualization
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../14_bigwig_generation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BigWig Generation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../15_visualization_heatmaps/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Visualization Heatmaps
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../16_chipseeker_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Annotation with ChIPseeker
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial_book_creation.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    How We Built This Book
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../CONTRIBUTORS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributors
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#basic-concept-the-puzzle" class="md-nav__link">
    <span class="md-ellipsis">
      
        Basic Concept (The "Puzzle")
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#execution-solving-it" class="md-nav__link">
    <span class="md-ellipsis">
      
        Execution (Solving It)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Execution (Solving It)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-the-index-building-the-map" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1: The Index (Building the Map)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#after-indexing-bowtie2" class="md-nav__link">
    <span class="md-ellipsis">
      
        After Indexing (Bowtie2)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-single-sample-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2: Single Sample Alignment
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-structure" class="md-nav__link">
    <span class="md-ellipsis">
      
        Output Structure
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-automation-loop" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3: Automation Loop
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optimization" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optimization
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Optimization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#optimization-threads-vs-jobs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Optimization: Threads vs. Jobs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-bam-file-structure" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding BAM File Structure
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multimapping-the-lost-gps" class="md-nav__link">
    <span class="md-ellipsis">
      
        Multimapping &amp; The "Lost GPS"
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-check-samtools-flagstat-samtools-stats" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Check: samtools flagstat &amp; samtools stats
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Quality Check: samtools flagstat &amp; samtools stats">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quick-manual-mapq-check" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Manual MAPQ Check
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-multi-mapping-reads" class="md-nav__link">
    <span class="md-ellipsis">
      
        Filtering Multi-mapping Reads
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#directory-structure-after-alignment-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directory Structure After Alignment QC
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        Summary
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="alignment-solving-the-jigsaw-puzzle">Alignment (Solving the Jigsaw Puzzle)<a class="headerlink" href="#alignment-solving-the-jigsaw-puzzle" title="Permanent link">&para;</a></h1>
<p><code>Bowtie2</code> <code>alignment</code> <code>BAM-files</code> <code>single-end</code> <code>paired-end</code> <code>samtools</code> <code>genome-index</code> <code>MAPQ</code> <code>quality-control</code> <code>multimapping</code> <code>ChIP-seq</code> <code>flagstat</code> <code>MultiQC</code> <code>filtering</code></p>
<h2 id="basic-concept-the-puzzle">Basic Concept (The "Puzzle")<a class="headerlink" href="#basic-concept-the-puzzle" title="Permanent link">&para;</a></h2>
<p>Imagine your Reference Genome is the <strong>Picture on the Puzzle Box</strong> (a complete image of the DNA).
Your Reads (FASTAS) are the millions of tiny <strong>Puzzle Pieces</strong> scattered on the floor.</p>
<p><strong>Alignment</strong> is simply picking up every piece and finding exactly where it fits on the picture.</p>
<ul>
<li><strong>The Input:</strong> Millions of jumbled reads.</li>
<li><strong>The Tool:</strong> <strong>Bowtie2</strong> (The Puzzle Solver).</li>
<li><strong>The Output:</strong> A <strong>BAM File</strong>. This is the digital record of where every piece belongs.</li>
</ul>
<hr />
<h2 id="execution-solving-it">Execution (Solving It)<a class="headerlink" href="#execution-solving-it" title="Permanent link">&para;</a></h2>
<h3 id="step-1-the-index-building-the-map">Step 1: The Index (Building the Map)<a class="headerlink" href="#step-1-the-index-building-the-map" title="Permanent link">&para;</a></h3>
<p>Before Bowtie2 can work, it needs to process the reference genome into a format it can search quickly. This is called an <strong>Index</strong>.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Example syntax: bowtie2-build [genome.fa] [prefix_name]</span>
bowtie2-build<span class="w"> </span>genome_index/ce.fa<span class="w"> </span>genome_index/ce_index<span class="w">            </span><span class="c1"># ce.fa = C. elegans reference genome</span>
</code></pre></div>
<p><em>You only do this once!</em></p>
<p>You can download the C. elegans reference genome from Ensembl using the following link:  <a href="https://ftp.ensembl.org/pub/release-115/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna.toplevel.fa.gz">C. elegans genome</a>.  Place it in the <strong>genome_index</strong> directory, decompress it, and rename the file to <code>ce.fa</code></p>
<h3 id="after-indexing-bowtie2">After Indexing (Bowtie2)<a class="headerlink" href="#after-indexing-bowtie2" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
├── fastq_raw/
│   └── ...
├── fastq_cleaned/                ← Fastp cleaned reads
│   └── ...
├── fastp_reports/
│   └── ...
├── genome_index/           ← Bowtie2 index files
│   ├── ce_index.1.bt2
│   ├── ce_index.2.bt2
│   ├── ce_index.3.bt2
│   ├── ce_index.4.bt2
│   ├── ce_index.rev.1.bt2
│   └── ce_index.rev.2.bt2
└── sample_id.txt
</code></pre></div>
<h3 id="step-2-single-sample-alignment">Step 2: Single Sample Alignment<a class="headerlink" href="#step-2-single-sample-alignment" title="Permanent link">&para;</a></h3>
<p><strong>Run Bowtie2 alignment for a single-end sample</strong></p>
<p><strong>Input files needed:</strong></p>
<ul>
<li>Reference genome index: <code>genome_index/ce_index</code> (created in previous step)</li>
<li>Cleaned FASTQ file: <code>fastq_cleaned/H3K27me3_IP_rep1.clean.fastq.gz</code></li>
</ul>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>bowalign

bowtie2<span class="w"> </span>-x<span class="w"> </span>genome_index/ce_index<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-U<span class="w"> </span>fastq_cleaned/H3K27me3_IP_rep1.clean.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-p<span class="w"> </span><span class="m">6</span><span class="w"> </span>--no-unal<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="m">2</span>&gt;<span class="w"> </span>bowalign/H3K27me3_IP_rep1.log<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">6</span><span class="w"> </span>-o<span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam

samtools<span class="w"> </span>index<span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam
</code></pre></div>
<p><strong>What this does:</strong></p>
<ol>
<li><strong>bowtie2</strong> aligns reads to the reference genome using the <code>-U</code> flag for single-end data</li>
<li><strong>-p 6</strong> uses 6 CPU threads for faster processing</li>
<li><strong>--no-unal</strong> suppresses unaligned reads from output (reads that don't match the genome are excluded, saving disk space and processing time downstream)</li>
<li><strong>2&gt;</strong> saves alignment statistics to a log file</li>
<li><strong>samtools sort</strong> sorts alignments by genomic position (required for downstream analysis)</li>
<li><strong>samtools index</strong> creates an index file (.bai) for fast random access</li>
</ol>
<blockquote>
<p>[!NOTE]
<strong>If you have paired-end data</strong>, use <code>-1</code> and <code>-2</code> flags instead of <code>-U</code>:</p>
</blockquote>
<div class="highlight"><pre><span></span><code>bowtie2<span class="w"> </span>-x<span class="w"> </span>genome_index/ce_index<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-1<span class="w"> </span>fastq_cleaned/Sample_R1.clean.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-2<span class="w"> </span>fastq_cleaned/Sample_R2.clean.fastq.gz<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-p<span class="w"> </span><span class="m">6</span><span class="w"> </span>--no-unal<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="m">2</span>&gt;<span class="w"> </span>bowalign/Sample.log<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">6</span><span class="w"> </span>-o<span class="w"> </span>bowalign/Sample.sorted.bam

samtools<span class="w"> </span>index<span class="w"> </span>bowalign/Sample.sorted.bam
</code></pre></div>
<p><strong>Final Touch:</strong>
We always "index" the BAM file. Think of this as creating a <strong>Table of Contents</strong> so the computer can jump to any chromosome instantly.</p>
<div class="highlight"><pre><span></span><code>samtools<span class="w"> </span>index<span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam
</code></pre></div>
<h3 id="output-structure"><strong>Output Structure</strong><a class="headerlink" href="#output-structure" title="Permanent link">&para;</a></h3>
<p>After running this step, your directory should look like:</p>
<div class="highlight"><pre><span></span><code>bowalign/
├── H3K27me3_IP_rep1.log
├── H3K27me3_IP_rep1.sorted.bam
└── H3K27me3_IP_rep1.sorted.bam.bai
</code></pre></div>
<p>Once this single run completes successfully, you can confidently automate for all samples.</p>
<h3 id="step-3-automation-loop">Step 3: Automation Loop<a class="headerlink" href="#step-3-automation-loop" title="Permanent link">&para;</a></h3>
<p>After creating <code>sample_id.txt</code> (see <a href="../03_geo_fastq_download/#24-creating-sample-id-list">Section 2.4</a>), here is the script to run this for all your samples:</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="nb">set</span><span class="w"> </span>-euo<span class="w"> </span>pipefail

mkdir<span class="w"> </span>-p<span class="w"> </span>bowalign<span class="w"> </span>bowalign_log

<span class="k">while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>-r<span class="w"> </span>sample<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Aligning </span><span class="nv">$sample</span><span class="s2">&quot;</span>

<span class="w">  </span>bowtie2<span class="w"> </span>-x<span class="w"> </span>genome_index/ce_index<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-U<span class="w"> </span><span class="s2">&quot;fastq_cleaned/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.clean.fastq.gz&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-p<span class="w"> </span><span class="m">6</span><span class="w"> </span>--no-unal<span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="m">2</span>&gt;<span class="w"> </span><span class="s2">&quot;bowalign_log/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.bowtie2.log&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">6</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;bowalign/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.sorted.bam&quot;</span>

<span class="w">  </span>samtools<span class="w"> </span>index<span class="w"> </span><span class="s2">&quot;bowalign/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.sorted.bam&quot;</span>

<span class="k">done</span><span class="w"> </span>&lt;<span class="w"> </span>sample_id.txt
</code></pre></div>
<blockquote>
<p>[!NOTE]
<strong>Paired-end while loop</strong> (if your samples have _R1 and_R2 files):</p>
</blockquote>
<div class="highlight"><pre><span></span><code><span class="k">while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>-r<span class="w"> </span>sample<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Aligning </span><span class="nv">$sample</span><span class="s2">&quot;</span>

<span class="w">  </span>bowtie2<span class="w"> </span>-x<span class="w"> </span>genome_index/ce_index<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-1<span class="w"> </span><span class="s2">&quot;fastq_cleaned/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">_R1.clean.fastq.gz&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-2<span class="w"> </span><span class="s2">&quot;fastq_cleaned/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">_R2.clean.fastq.gz&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-p<span class="w"> </span><span class="m">6</span><span class="w"> </span>--no-unal<span class="w"> </span><span class="se">\</span>
<span class="w">    </span><span class="m">2</span>&gt;<span class="w"> </span><span class="s2">&quot;bowalign_log/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.log&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">6</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;bowalign/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.sorted.bam&quot;</span>

<span class="w">  </span>samtools<span class="w"> </span>index<span class="w"> </span><span class="s2">&quot;bowalign/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.sorted.bam&quot;</span>

<span class="k">done</span><span class="w"> </span>&lt;<span class="w"> </span>sample_id.txt
</code></pre></div>
<hr />
<h2 id="optimization">Optimization<a class="headerlink" href="#optimization" title="Permanent link">&para;</a></h2>
<h3 id="optimization-threads-vs-jobs">Optimization: Threads vs. Jobs<a class="headerlink" href="#optimization-threads-vs-jobs" title="Permanent link">&para;</a></h3>
<p>You have a limited number of CPU cores (computers brains). You can use them in two ways:</p>
<ol>
<li><strong>Multi-Threading (<code>-p 6</code>):</strong> One sample uses 6 cores. It finishes very fast, but you only do <strong>one sample at a time</strong>.<ul>
<li><em>Best for:</em> Large genomes, low memory.</li>
</ul>
</li>
<li><strong>Parallel Jobs:</strong> You run <strong>3 samples</strong> at once, and each sample uses <strong>2 cores</strong>.<ul>
<li><em>Best for:</em> Many small samples (RNA-seq, small genomes).</li>
</ul>
</li>
</ol>
<p><strong>Rule of Thumb:</strong> <code>bowtie2</code> stops getting faster after about <strong>8 threads</strong>. Don't give it 50 threads; it's a waste!</p>
<p><a href="https://www.jefftk.com/p/benchmarking-bowtie2-threading">Benchmarking Bowtie2 Threading - Jeff Kaufman (2023)</a></p>
<p><a href="https://wiki.csi.cuny.edu/HPCCWiki/BOWTIE2">BOWTIE2 - HPCC Wiki</a></p>
<p><a href="https://github.com/samtools/samtools/issues">Guidance with using multiple threads with samtools - GitHub</a></p>
<p><strong>Example with GNU Parallel</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="nb">set</span><span class="w"> </span>-euo<span class="w"> </span>pipefail

mkdir<span class="w"> </span>-p<span class="w"> </span>bowalign<span class="w"> </span>bowalign_log

parallel<span class="w"> </span>-j<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="s1">&#39;</span>
<span class="s1">  bowtie2 \</span>
<span class="s1">    -x genome_index/ce_index \</span>
<span class="s1">    -U fastq_cleaned/{}.clean.fastq.gz \</span>
<span class="s1">    -p 4 \</span>
<span class="s1">    --no-unal \</span>
<span class="s1">    2&gt; bowalign_log/{}.log \</span>
<span class="s1">  | samtools sort \</span>
<span class="s1">      -@ 2 \</span>
<span class="s1">      -m 1G \</span>
<span class="s1">      -o bowalign/{}.sorted.bam</span>

<span class="s1">  samtools index bowalign/{}.sorted.bam</span>
<span class="s1">&#39;</span><span class="w"> </span>::::<span class="w"> </span>sample_id.txt
</code></pre></div>
<p><strong>Effective CPU usage (implied by this setup):</strong></p>
<ul>
<li>2 parallel samples (<code>-j 2</code>)</li>
<li>Per sample: bowtie2 (4 threads, <code>-p 4</code>) + samtools sort (2 threads, <code>-@ 2</code>)</li>
<li>Total ≈ 12 threads → safe on a 16-core machine</li>
</ul>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
├── fastq_raw/
├── fastq_cleaned/
├── fastp_reports/
├── genome_index/
├── bowalign/                ← Aligned BAM files
│   ├── H3K27me3_IP_rep1.sorted.bam
│   ├── H3K27me3_IP_rep1.sorted.bam.bai
│   └── ...
├── bowalign_log/            ← Alignment statistics
│   ├── H3K27me3_IP_rep1.log
│   └── ...
└── sample_id.txt
</code></pre></div>
<h3 id="understanding-bam-file-structure">Understanding BAM File Structure<a class="headerlink" href="#understanding-bam-file-structure" title="Permanent link">&para;</a></h3>
<p>Before we check alignment quality, let's look at what's inside a BAM file. BAM files store alignment information in a structured format with <strong>11 mandatory columns</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># View the first few alignments in human-readable format</span>
samtools<span class="w"> </span>view<span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-3
</code></pre></div>
<p><strong>Example output (single-end data):</strong></p>
<div class="highlight"><pre><span></span><code>SRR7297994.1    0    I      15200    42    76M    *    0    0    ACGTACGT...    IIIIIIII...
SRR7297994.2    0    II     98432    30    76M    *    0    0    TGCATGCA...    IIIHHHHH...
SRR7297994.3   16    III    45123     0    76M    *    0    0    GATTACA...     IIIHHHGG...
</code></pre></div>
<p><strong>Column Breakdown (First 5 of 11 columns):</strong></p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Name</th>
<th>Example</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>QNAME</td>
<td>SRR7297994.1</td>
<td>Read/Query name</td>
</tr>
<tr>
<td>2</td>
<td><strong>FLAG</strong></td>
<td>0, 16</td>
<td><strong>Bitwise FLAG</strong> (flagstat reads this!)</td>
</tr>
<tr>
<td>3</td>
<td>RNAME</td>
<td>I, II, III</td>
<td>Reference sequence name (chromosome)</td>
</tr>
<tr>
<td>4</td>
<td>POS</td>
<td>15200</td>
<td>Leftmost mapping position</td>
</tr>
<tr>
<td>5</td>
<td><strong>MAPQ</strong></td>
<td>42, 30, 0</td>
<td><strong>Mapping Quality score</strong> (0-255)</td>
</tr>
</tbody>
</table>
<p><strong>What Each QC Tool Evaluates:</strong></p>
<p><strong><code>samtools flagstat</code></strong> reads <strong>Column 2 (FLAG)</strong>:</p>
<ul>
<li>Bit 4: Read unmapped?</li>
<li>Bit 256: Secondary alignment?</li>
<li>Bit 1024: PCR/optical duplicate?</li>
<li>Bit 2048: Supplementary alignment?</li>
<li>And more... (see FLAG section below)</li>
</ul>
<blockquote>
<p>[!TIP]
<strong>Decode FLAGS:</strong> Use <a href="https://broadinstitute.github.io/picard/explain-flags.html">Explain SAM Flags</a> to understand bitwise FLAG values.</p>
</blockquote>
<h3 id="multimapping-the-lost-gps">Multimapping &amp; The "Lost GPS"<a class="headerlink" href="#multimapping-the-lost-gps" title="Permanent link">&para;</a></h3>
<p>Sometimes a read is repetitive (e.g., "ATATATAT"). It fits in 50 different places on the genome.
The aligner doesn't know which spot is correct, so it gives it a <strong>Low MAPQ Score</strong> (Mapping Quality).
<strong>MAPQ filtering</strong> (<code>-q 30</code>) reads <strong>Column 5 (MAPQ)</strong>:</p>
<ul>
<li><strong>MAPQ = 0:</strong> "I have NO clue where this goes. It fits in many places." (Multimapped)</li>
<li><strong>MAPQ &gt; 30:</strong> "I am highly confident this read belongs EXACTLY here." (Unique)</li>
</ul>
<h3 id="quality-check-samtools-flagstat-samtools-stats">Quality Check: <code>samtools flagstat</code> &amp; <code>samtools stats</code><a class="headerlink" href="#quality-check-samtools-flagstat-samtools-stats" title="Permanent link">&para;</a></h3>
<p>Did the alignment work? Let's check the score using two tools:</p>
<ul>
<li><strong>flagstat</strong>: Quick alignment summary (mapped reads, pairs, etc.)</li>
<li><strong>stats</strong>: Comprehensive metrics including MAPQ distribution (for MultiQC visualization)</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># Create QC output directory for alignment metrics</span>
mkdir<span class="w"> </span>-p<span class="w"> </span>bowalign_qc

<span class="c1"># Quick summary with flagstat</span>
samtools<span class="w"> </span>flagstat<span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>bowalign_qc/H3K27me3_IP_rep1.flagstat.txt

<span class="c1"># Comprehensive stats (includes MAPQ distribution for MultiQC)</span>
samtools<span class="w"> </span>stats<span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>bowalign_qc/H3K27me3_IP_rep1.stats.txt
</code></pre></div>
<p><strong>Batch processing for all samples:</strong></p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="nb">set</span><span class="w"> </span>-euo<span class="w"> </span>pipefail

mkdir<span class="w"> </span>-p<span class="w"> </span>bowalign_qc

<span class="k">while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>-r<span class="w"> </span>sample<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Running QC for: </span><span class="nv">$sample</span><span class="s2">&quot;</span>

<span class="w">  </span><span class="c1"># Generate flagstat summary</span>
<span class="w">  </span>samtools<span class="w"> </span>flagstat<span class="w"> </span>bowalign/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>bowalign_qc/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.flagstat.txt

<span class="w">  </span><span class="c1"># Generate comprehensive stats (for MultiQC)</span>
<span class="w">  </span>samtools<span class="w"> </span>stats<span class="w"> </span>bowalign/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>bowalign_qc/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.stats.txt

<span class="k">done</span><span class="w"> </span>&lt;<span class="w"> </span>sample_id.txt

<span class="c1"># Generate MultiQC report with all QC results</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Generating MultiQC report...&quot;</span>
multiqc<span class="w"> </span>bowalign_qc/<span class="w"> </span>-o<span class="w"> </span>bowalign_qc/<span class="w"> </span>-n<span class="w"> </span>alignment_qc_report
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;MultiQC report saved: bowalign_qc/alignment_qc_report.html&quot;</span>
</code></pre></div>
<p>This creates an interactive HTML report (<code>alignment_qc_report.html</code>) with visualizations of alignment metrics across all samples, including MAPQ distribution plots.</p>
<hr />
<p><strong>What to look for:</strong></p>
<ul>
<li><strong>Mapping Rate:</strong> Ideally <strong>&gt;80-90%</strong>. If it's &lt;50%, your DNA might be contaminated (e.g., bacteria in a sample).</li>
<li><strong>Goal:</strong> High mapping percentage indicates good quality alignment.</li>
<li><strong>Warning:</strong> If mapping is low (&lt;50%), you may have the wrong organism or bad sequencing.</li>
</ul>
<p><strong>The Sensitivity-Specificity Trade-off:</strong></p>
<p>Multi-mapping reads critically influence the balance between sensitivity and specificity in ChIP-seq peak detection. Excluding multi-mappers, standard practice in most peak callers, improves specificity by preventing artificial signal inflation in repetitive regions but reduces sensitivity for genuine binding events within those regions (<a href="https://doi.org/10.1093/bib/bbw023">Nakato et al., 2016</a>). This trade-off is acceptable for transcription factors and chromatin features predominantly in unique genomic loci. However, repetitive and transposable elements (REs/TEs) constitute significant genome portions with important regulatory roles, and discarding multi-mapped reads substantially underrepresents regulatory events in these regions (<a href="https://doi.org/10.1101/gr.278638.123">Morrissey et al., 2024</a>).</p>
<h4 id="quick-manual-mapq-check">Quick Manual MAPQ Check<a class="headerlink" href="#quick-manual-mapq-check" title="Permanent link">&para;</a></h4>
<p>For a <strong>quick manual inspection</strong> of MAPQ distribution (complementary to the MultiQC visualization):</p>
<div class="highlight"><pre><span></span><code>samtools<span class="w"> </span>view<span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;{print $5}&#39;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sort<span class="w"> </span>-n<span class="w"> </span><span class="p">|</span><span class="w"> </span>uniq<span class="w"> </span>-c<span class="w"> </span>&gt;<span class="w"> </span>bowalign_qc/H3K27me3_IP_rep1.mapq.txt
</code></pre></div>
<p><strong>Understanding the Manual Output:</strong></p>
<ul>
<li><strong>If you see lots of 0s:</strong> Your file includes "Lost GPS" reads (Multimappers).</li>
<li><strong>If your scores start at 30+:</strong> Your file has <strong>Already Filtered</strong> the bad reads.</li>
</ul>
<p><strong>Output format: <code>count</code> <code>MAPQ_score</code></strong></p>
<div class="highlight"><pre><span></span><code>123964  30  &lt;-- Lowest score is 30 (Good confidence)
1928 31
20741 32
1477 33
4040 34
34434 35
14294 36
53329 37
30665 38
88528 39
77123 40
2960636 42  &lt;-- Highest score is 42 (Perfect confidence)
</code></pre></div>
<p><em>Verdict: This BAM file contains only uniquely mapped, high-quality reads.</em></p>
<hr />
<h3 id="filtering-multi-mapping-reads">Filtering Multi-mapping Reads<a class="headerlink" href="#filtering-multi-mapping-reads" title="Permanent link">&para;</a></h3>
<p>For standard ChIP-seq analysis (TFs, histone marks in unique regions), filter BAM files to retain only high-quality uniquely mapped reads:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Create directory for filtered BAM files</span>
mkdir<span class="w"> </span>-p<span class="w"> </span>bowalign_filtered

<span class="c1"># Filter out multi-mappers (keep only MAPQ ≥ 30)</span>
samtools<span class="w"> </span>view<span class="w"> </span>-b<span class="w"> </span>-q<span class="w"> </span><span class="m">30</span><span class="w"> </span>bowalign/H3K27me3_IP_rep1.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>bowalign_filtered/H3K27me3_IP_rep1.filtered.bam

<span class="c1"># Index the filtered BAM</span>
samtools<span class="w"> </span>index<span class="w"> </span>bowalign_filtered/H3K27me3_IP_rep1.filtered.bam
</code></pre></div>
<p><strong>Batch processing for all samples:</strong></p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="nb">set</span><span class="w"> </span>-euo<span class="w"> </span>pipefail

mkdir<span class="w"> </span>-p<span class="w"> </span>bowalign_filtered

<span class="k">while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>-r<span class="w"> </span>sample<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Filtering multi-mappers for: </span><span class="nv">$sample</span><span class="s2">&quot;</span>
<span class="w">  </span>samtools<span class="w"> </span>view<span class="w"> </span>-b<span class="w"> </span>-q<span class="w"> </span><span class="m">30</span><span class="w"> </span>bowalign/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.sorted.bam<span class="w"> </span>&gt;<span class="w"> </span>bowalign_filtered/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.filtered.bam
<span class="w">  </span>samtools<span class="w"> </span>index<span class="w"> </span>bowalign_filtered/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.filtered.bam
<span class="k">done</span><span class="w"> </span>&lt;<span class="w"> </span>sample_id.txt
</code></pre></div>
<blockquote>
<p>[!NOTE]
The <code>-q 30</code> flag filters out reads with MAPQ &lt; 30 (removes multi-mappers and low-confidence alignments). Most downstream tools (MACS2, deepTools) can also filter by MAPQ, so this step is optional but recommended for cleaner data.</p>
</blockquote>
<hr />
<h3 id="directory-structure-after-alignment-qc">Directory Structure After Alignment QC<a class="headerlink" href="#directory-structure-after-alignment-qc" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
├── fastq_raw/
├── fastq_cleaned/
├── genome_index/
├── bowalign/                    ← Aligned BAM files
│   ├── H3K27me3_IP_rep1.sorted.bam
│   ├── H3K27me3_IP_rep1.sorted.bam.bai
│   └── ...
├── bowalign_log/                ← Alignment statistics
│   ├── H3K27me3_IP_rep1.log
│   └── ...
├── bowalign_qc/                 ← Quality control metrics
│   ├── H3K27me3_IP_rep1.flagstat.txt
│   ├── H3K27me3_IP_rep1.stats.txt
│   ├── H3K27me3_IP_rep1.mapq.txt
│   ├── alignment_qc_report.html
│   ├── alignment_qc_report_data/
│   └── ...
├── bowalign_filtered/           ← Filtered BAM files (MAPQ ≥ 30)
│   ├── H3K27me3_IP_rep1.filtered.bam
│   ├── H3K27me3_IP_rep1.filtered.bam.bai
│   └── ...
└── sample_id.txt
</code></pre></div>
<hr />
<h2 id="summary">Summary<a class="headerlink" href="#summary" title="Permanent link">&para;</a></h2>
<ol>
<li><strong>Analogy:</strong> Alignment is placing puzzle pieces onto the reference picture.</li>
<li><strong>Action:</strong> Use <code>bowtie2</code> to align and <code>samtools sort</code> to organize.</li>
<li><strong>Result:</strong> A <strong>Sorted BAM</strong> file (the solved puzzle), ready for peak calling.</li>
</ol>
<blockquote>
<p>[!NOTE]
<strong>Up Next:</strong> Before peak calling, we need to remove PCR duplicates and assess library quality.</p>
</blockquote>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 Raja Ishaq Nabi Khan
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/ishaaq34" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.top", "search.suggest", "search.highlight", "content.code.copy"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>