
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A comprehensive, hands-on tutorial for ChIP-seq data analysis">
      
      
        <meta name="author" content="Raja Ishaq Nabi Khan">
      
      
        <link rel="canonical" href="https://ishaaq34.github.io/Chipseq_analysis_tutorial_mkdocs/14_bigwig_generation/">
      
      
        <link rel="prev" href="../13_idr_consensus_motifs_rk_corrected/">
      
      
        <link rel="next" href="../15_visualization_heatmaps/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>BigWig Generation - ChIP-seq Analysis Tutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bam-to-bigwig-visualizing-the-signal" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-header__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq Analysis Tutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              BigWig Generation
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../00_introduction/" class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../01_setup_environment/" class="md-tabs__link">
          
  
  
  Setup & Data

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../05_alignment_bowtie2/" class="md-tabs__link">
          
  
  
  Alignment & QC

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../11_macs3_peak_calling/" class="md-tabs__link">
          
  
  
  Peak Calling

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="./" class="md-tabs__link">
          
  
  
  Visualization

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../tutorial_book_creation/" class="md-tabs__link">
          
  
  
  Appendix

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-nav__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    ChIP-seq Analysis Tutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Setup & Data
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Setup & Data
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_setup_environment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Environment Setup
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_bash_automation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Bash Automation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_geo_fastq_download/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GEO Fastq Download
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_fastq_concepts/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fastq Concepts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Alignment & QC
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Alignment & QC
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05_alignment_bowtie2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Alignment with Bowtie2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06_duplicate_removal_qc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Duplicate Removal & QC
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_library_complexity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Library Complexity
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../08_bam_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Quality Metrics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../09_strand_cross_correlation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Strand Cross Correlation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../10_bam_summary_fingerprint/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Summary & Fingerprint
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Peak Calling
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Peak Calling
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../11_macs3_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MACS3 Peak Calling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../12_frip_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Metrics - FRiP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../13_idr_consensus_motifs_rk_corrected/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Visualization
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Visualization
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    BigWig Generation
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    BigWig Generation
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-basic-concept-the-traffic-map" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Basic Concept (The Traffic Map)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Basic Concept (The Traffic Map)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#why-bigwig" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why BigWig?
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-requirements-effective-genome-size" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Requirements (Effective Genome Size)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Requirements (Effective Genome Size)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#how-we-calculated-it" class="md-nav__link">
    <span class="md-ellipsis">
      
        How we calculated it
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="How we calculated it">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#method-1-fasize-calculates-general-effective-genome" class="md-nav__link">
    <span class="md-ellipsis">
      
        Method 1: faSize (Calculates General Effective Genome)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#method-2-khmer-calculates-unique-effective-genome" class="md-nav__link">
    <span class="md-ellipsis">
      
        Method 2: khmer (Calculates Unique Effective Genome)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-result" class="md-nav__link">
    <span class="md-ellipsis">
      
        The Result
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-decision" class="md-nav__link">
    <span class="md-ellipsis">
      
        The Decision
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-execution-the-converter" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Execution (The Converter)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Execution (The Converter)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-31-create-output-directory" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3.1: Create Output Directory
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-32-run-bamcoverage-loop" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3.2: Run bamCoverage Loop
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-fine-tuning-under-the-hood" class="md-nav__link">
    <span class="md-ellipsis">
      
        4. Fine Tuning (Under the Hood)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="4. Fine Tuning (Under the Hood)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-bin-size-resolution" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.1 Bin Size (Resolution)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#42-smoothing-blurring-the-photo" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.2 Smoothing (Blurring the Photo)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#43-normalization-the-currency-exchange" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.3 Normalization (The Currency Exchange)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        Summary
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#directory-structure-after-bigwig-generation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directory Structure After BigWig Generation
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../15_visualization_heatmaps/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Visualization Heatmaps
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../16_chipseeker_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Annotation with ChIPseeker
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Appendix
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Appendix
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial_book_creation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    How We Built This Book
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../CONTRIBUTORS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributors
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-basic-concept-the-traffic-map" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Basic Concept (The Traffic Map)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Basic Concept (The Traffic Map)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#why-bigwig" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why BigWig?
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-requirements-effective-genome-size" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Requirements (Effective Genome Size)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Requirements (Effective Genome Size)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#how-we-calculated-it" class="md-nav__link">
    <span class="md-ellipsis">
      
        How we calculated it
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="How we calculated it">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#method-1-fasize-calculates-general-effective-genome" class="md-nav__link">
    <span class="md-ellipsis">
      
        Method 1: faSize (Calculates General Effective Genome)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#method-2-khmer-calculates-unique-effective-genome" class="md-nav__link">
    <span class="md-ellipsis">
      
        Method 2: khmer (Calculates Unique Effective Genome)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-result" class="md-nav__link">
    <span class="md-ellipsis">
      
        The Result
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-decision" class="md-nav__link">
    <span class="md-ellipsis">
      
        The Decision
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-execution-the-converter" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Execution (The Converter)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Execution (The Converter)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-31-create-output-directory" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3.1: Create Output Directory
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-32-run-bamcoverage-loop" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3.2: Run bamCoverage Loop
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-fine-tuning-under-the-hood" class="md-nav__link">
    <span class="md-ellipsis">
      
        4. Fine Tuning (Under the Hood)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="4. Fine Tuning (Under the Hood)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-bin-size-resolution" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.1 Bin Size (Resolution)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#42-smoothing-blurring-the-photo" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.2 Smoothing (Blurring the Photo)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#43-normalization-the-currency-exchange" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.3 Normalization (The Currency Exchange)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        Summary
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#directory-structure-after-bigwig-generation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directory Structure After BigWig Generation
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="bam-to-bigwig-visualizing-the-signal">Bam to BigWig (Visualizing the Signal)<a class="headerlink" href="#bam-to-bigwig-visualizing-the-signal" title="Permanent link">&para;</a></h1>
<p><code>bamCoverage</code> <code>bigWig</code> <code>normalization</code> <code>RPGC</code> <code>effective-genome-size</code> <code>visualization</code></p>
<h2 id="1-basic-concept-the-traffic-map">1. Basic Concept (The Traffic Map)<a class="headerlink" href="#1-basic-concept-the-traffic-map" title="Permanent link">&para;</a></h2>
<h3 id="why-bigwig">Why BigWig?<a class="headerlink" href="#why-bigwig" title="Permanent link">&para;</a></h3>
<hr />
<p>A <strong>BAM</strong> file gives you the location of every single "car" (read) on the road. It's massive and slow to load.</p>
<p>A <strong>BigWig</strong> file is like the <strong>Google Maps Traffic View</strong>. It doesn't show you the individual cars; it just shows you a green, yellow, or red line indicating "Volume".</p>
<ul>
<li><strong>Small &amp; Fast:</strong> Compact file size.</li>
<li><strong>Visual:</strong> Perfect for viewing on a Genome Browser (IGV/UCSC).</li>
</ul>
<hr />
<h2 id="2-requirements-effective-genome-size">2. Requirements (Effective Genome Size)<a class="headerlink" href="#2-requirements-effective-genome-size" title="Permanent link">&para;</a></h2>
<p>The "Effective" size is the part of the genome that is mappable.</p>
<ul>
<li><strong>Unique Regions:</strong> Easy to map.</li>
<li><strong>Repetitive Regions:</strong> Hard to map, but reads can technically align there (Ambiguous).</li>
</ul>
<p><strong>Formula 1: Total Raw Length</strong>
$$
\text{Total Genome Length} = \underbrace{\text{Effective Genome}}<em _text_Gaps_Unsequenced="\text{Gaps/Unsequenced">{\text{All Mappable}} + \underbrace{\text{N-bases}}</em>
$$}</p>
<p><strong>Formula 2: Breaking Down "Effective Genome"</strong>
$$
\text{Effective Genome} = \underbrace{\text{Unique Genome}}<em _text_Ambiguous="\text{Ambiguous">{\text{Easy}} + \underbrace{\text{Repetitive DNA}}</em>
$$}</p>
<h3 id="how-we-calculated-it">How we calculated it<a class="headerlink" href="#how-we-calculated-it" title="Permanent link">&para;</a></h3>
<p>There are <strong>two main ways</strong> to estimate this:</p>
<h4 id="method-1-fasize-calculates-general-effective-genome">Method 1: faSize (Calculates General Effective Genome)<a class="headerlink" href="#method-1-fasize-calculates-general-effective-genome" title="Permanent link">&para;</a></h4>
<p>The <strong>faSize</strong> tool counts all non-N bases. This assumes that if a read aligns to a repeat, it still counts as "mapped".</p>
<p><strong>Step 1.1: Extract chr11 and chr12</strong></p>
<div class="highlight"><pre><span></span><code>samtools<span class="w"> </span>faidx<span class="w"> </span>genome.fasta<span class="w"> </span>chr11<span class="w"> </span>chr12<span class="w"> </span>&gt;<span class="w"> </span>chr11_chr12.fasta
</code></pre></div>
<p><strong>Step 1.2: Get statistics</strong></p>
<div class="highlight"><pre><span></span><code>faSize<span class="w"> </span>chr11_chr12.fasta
</code></pre></div>
<p><strong>Output:</strong></p>
<div class="highlight"><pre><span></span><code>268361931 bases (690373 N&#39;s 267671558 real 267671558 upper 0 lower) in 2 sequences inside 1 file
</code></pre></div>
<p><strong>Step 1.3: Calculate total effective size (non-N)</strong></p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;{nonN = $2 - $5; sum += nonN} END {print sum}&#39;</span><span class="w"> </span>chr11_chr11_chr12.faSize.txt
<span class="c1"># Result: 268361931</span>
</code></pre></div>
<p><strong>Result:</strong> <code>268,361,931 bp</code> (Unique + Repetitive)</p>
<h4 id="method-2-khmer-calculates-unique-effective-genome">Method 2: khmer (Calculates Unique Effective Genome)<a class="headerlink" href="#method-2-khmer-calculates-unique-effective-genome" title="Permanent link">&para;</a></h4>
<p>The <strong>khmer</strong> tool counts only unique k-mers. It strictly ignores repetitive zones.</p>
<p><strong>Command:</strong></p>
<div class="highlight"><pre><span></span><code>unique-kmers.py<span class="w"> </span>-k<span class="w"> </span><span class="m">21</span><span class="w"> </span>chr11_chr12.fasta
</code></pre></div>
<p><strong>Output:</strong></p>
<div class="highlight"><pre><span></span><code>(chip) rajaishaqnabikhan@Mac human % unique-kmers.py -k 21 chr11_chr12.fasta

|| This is the script unique-kmers.py in khmer.
|| You are running khmer version 3.0.0a3
|| You are also using screed version 1.1.3
||
|| If you use this script in a publication, please cite EACH of the following:
||
||   * MR Crusoe et al., 2015. https://doi.org/10.12688/f1000research.6924.1
||   * A. Döring et al. https://doi.org:80/10.1186/1471-2105-9-11
||   * Irber and Brown. https://doi.org/10.1101/056846
||
|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.

Estimated number of unique 21-mers in chr11_chr12.fasta: 220798375
Total estimated number of unique 21-mers: 220798375
</code></pre></div>
<h3 id="the-result">The Result<a class="headerlink" href="#the-result" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Raw length (Method 1: faSize):</strong> <code>268,361,931 bp</code> (Unique + Repetitive)</li>
<li><strong>Unique length (Method 2: khmer):</strong> <code>220,798,375 bp</code> (Unique Only)</li>
<li><strong>The Difference:</strong> <code>47,563,556 bp</code> of repetitive/unmappable DNA (Ambiguous).</li>
</ul>
<h3 id="the-decision">The Decision<a class="headerlink" href="#the-decision" title="Permanent link">&para;</a></h3>
<p>Which number do we use?</p>
<ul>
<li>Therefore, we must use the <strong>khmer</strong> result (<code>Unique Only</code>), because our data does not contain reads mapped to repetitive regions.</li>
</ul>
<blockquote>
<p>[!TIP]
<strong>Use this number:</strong> <code>220798375</code></p>
</blockquote>
<hr />
<h2 id="3-execution-the-converter">3. Execution (The Converter)<a class="headerlink" href="#3-execution-the-converter" title="Permanent link">&para;</a></h2>
<p>We use <code>bamCoverage</code> to convert BAM files into BigWig tracks. We will normalize using <strong>RPGC</strong> (Reads Per Genome Coverage) so all tracks are comparable (1x coverage scale).</p>
<h3 id="step-31-create-output-directory">Step 3.1: Create Output Directory<a class="headerlink" href="#step-31-create-output-directory" title="Permanent link">&para;</a></h3>
<p>Keep your workspace clean.</p>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>bigwigs
</code></pre></div>
<h3 id="step-32-run-bamcoverage-loop">Step 3.2: Run bamCoverage Loop<a class="headerlink" href="#step-32-run-bamcoverage-loop" title="Permanent link">&para;</a></h3>
<p>We process all BAM files using our <code>sample_id.txt</code> list.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Loop through each sample ID in the text file</span>
cat<span class="w"> </span>sample_id.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="k">while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>id<span class="p">;</span><span class="w"> </span><span class="k">do</span>

<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Generating BigWig for: </span><span class="nv">$id</span><span class="s2">&quot;</span>

<span class="w">  </span>bamCoverage<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-b<span class="w"> </span>encode_bam/<span class="si">${</span><span class="nv">id</span><span class="si">}</span>.bam<span class="w"> </span><span class="se">\ </span><span class="w">                   </span>
<span class="w">    </span>-o<span class="w"> </span>bigwigs/<span class="si">${</span><span class="nv">id</span><span class="si">}</span>.bw<span class="w"> </span><span class="se">\ </span><span class="w">                       </span>
<span class="w">    </span>--binSize<span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="se">\ </span><span class="w">                              </span>
<span class="w">    </span>--normalizeUsing<span class="w"> </span>RPGC<span class="w"> </span><span class="se">\ </span><span class="w">                     </span>
<span class="w">    </span>--effectiveGenomeSize<span class="w"> </span><span class="m">2701495761</span><span class="w"> </span><span class="se">\ </span><span class="w">           </span>
<span class="w">    </span>--smoothLength<span class="w"> </span><span class="m">30</span><span class="w"> </span><span class="se">\ </span><span class="w">                         </span>
<span class="w">    </span>--numberOfProcessors<span class="w"> </span><span class="m">4</span><span class="w"> </span>

<span class="k">done</span>
</code></pre></div>
<p><strong>What this does:</strong></p>
<ol>
<li><strong>Reads</strong> the BAM file.</li>
<li><strong>Chops</strong> the genome into 10bp bins (buckets).</li>
<li><strong>Counts</strong> reads in each bin.</li>
<li><strong>Normalizes</strong> the count so 1.0 = 1x genome coverage.</li>
<li><strong>Smooths</strong> the signal (averaging neighbors) to make the peaks look cleaner.</li>
<li><strong>Saves</strong> the result as a <code>.bw</code> file.</li>
</ol>
<hr />
<h2 id="4-fine-tuning-under-the-hood">4. Fine Tuning (Under the Hood)<a class="headerlink" href="#4-fine-tuning-under-the-hood" title="Permanent link">&para;</a></h2>
<h3 id="41-bin-size-resolution">4.1 Bin Size (Resolution)<a class="headerlink" href="#41-bin-size-resolution" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Concept:</strong> Think of this as the <strong>Resolution</strong> of your image.</li>
<li><strong>Small Bin (10bp):</strong> "HD" resolution. You see every tiny peak, but the file is larger and slower to compute.</li>
<li><strong>Large Bin (100bp):</strong> "SD" resolution. Good for zooming out and looking at broad trends. Faster to process.</li>
<li><strong>Decision:</strong> For transcription factors (sharp peaks), 10bp is great. For broad histone marks (H3K27me3), 50-100bp is often sufficient. We use <strong>10bp</strong> here for high detail.</li>
</ul>
<h3 id="42-smoothing-blurring-the-photo">4.2 Smoothing (Blurring the Photo)<a class="headerlink" href="#42-smoothing-blurring-the-photo" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Concept:</strong> Smoothing averages the signal of nearby bins to reduce "jumpy" noise.</li>
<li><strong>Why?</strong> Raw sequencing data can be pixelated. Smoothing applies a slight blur to make the biological signal (the hill) stand out against the background noise (the grass).</li>
<li><strong>Our Setting:</strong> <code>--smoothLength 30</code>. This averages the signal over a 30bp window.</li>
</ul>
<h3 id="43-normalization-the-currency-exchange">4.3 Normalization (The Currency Exchange)<a class="headerlink" href="#43-normalization-the-currency-exchange" title="Permanent link">&para;</a></h3>
<p>Samples have different sequencing depths.</p>
<ul>
<li><strong>Sample A:</strong> 40 million reads (Rich)</li>
<li><strong>Sample B:</strong> 20 million reads (Poor)</li>
</ul>
<p>If we don't fix this, Sample A will look huge just because it has more money (reads). <strong>Normalization</strong> puts everyone on the same currency.</p>
<ul>
<li><strong>RPGC (Reads Per Genome Coverage):</strong> <em>Preferred for ChIP-seq.</em></li>
<li><strong>Logic:</strong> "What would this signal look like if we had exactly <strong>1x coverage</strong> of the genome?"</li>
<li><strong>Why:</strong> It creates a standardized "Currency" (1x coverage) that makes biological sense. A value of "5.0" in the track means "5 times more reads than random background".</li>
</ul>
<hr />
<h2 id="summary">Summary<a class="headerlink" href="#summary" title="Permanent link">&para;</a></h2>
<ol>
<li><strong>Input:</strong> BAM files (<code>encode_bam/</code>).</li>
<li><strong>Action:</strong> <code>bamCoverage</code> with <strong>RPGC</strong> normalization.</li>
<li><strong>Output:</strong> BigWig files (<code>bigwigs/</code>) ready for IGV visualization.</li>
</ol>
<blockquote>
<p>[!NOTE]
<strong>Up Next:</strong> Now that we have our signals (BigWigs) and our QC (Fingerprints) done, we are ready to call peaks! (Wait, technically we usually call peaks <em>before</em> or <em>parallel</em> to visualization, but viewing tracks helps confirm peak calls).</p>
</blockquote>
<hr />
<h2 id="directory-structure-after-bigwig-generation">Directory Structure After BigWig Generation<a class="headerlink" href="#directory-structure-after-bigwig-generation" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
├── bam_files_final/            ← Source BAM files
├── macs3_results/              ← MACS3 peak files
├── idr/                        ← IDR outputs
└── bigwigs/                    ← **NEW: Normalized BigWig tracks**
    ├── H3K9ac_ENCFF534IPX.bw
    ├── H3K9ac_ENCFF193NPE.bw
    ├── H3K27me3_ENCFF164ALR.bw
    ├── H3K27me3_ENCFF532DQH.bw
    ├── Input_ENCFF110SOB.bw
    └── Input_ENCFF919XCV.bw
</code></pre></div>
<hr />












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["navigation.tabs", "navigation.sections", "navigation.top", "navigation.tracking", "search.suggest", "search.highlight", "content.code.copy"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>