
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A comprehensive, hands-on tutorial for ChIP-seq data analysis">
      
      
        <meta name="author" content="Raja Ishaq Nabi Khan">
      
      
        <link rel="canonical" href="https://ishaaq34.github.io/Chipseq_analysis_tutorial_mkdocs/13_idr_consensus_motifs_rk_corrected/">
      
      
        <link rel="prev" href="../12_frip_quality_metrics/">
      
      
        <link rel="next" href="../14_bigwig_generation/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>Reproducibility Analysis - ChIP-seq Analysis Tutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue-grey" data-md-color-accent="cyan">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#reproducibility-analysis-idr-consensus-peaks-motif-discovery" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-header__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq Analysis Tutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Reproducibility Analysis
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="blue-grey" data-md-color-accent="cyan"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="blue-grey" data-md-color-accent="cyan"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../00_introduction/" class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../01_setup_environment/" class="md-tabs__link">
          
  
  
  Setup & Data

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../05_alignment_bowtie2/" class="md-tabs__link">
          
  
  
  Alignment & QC

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../11_macs3_peak_calling/" class="md-tabs__link">
          
  
  
  Peak Calling

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../14_bigwig_generation/" class="md-tabs__link">
          
  
  
  Visualization

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../tutorial_book_creation/" class="md-tabs__link">
          
  
  
  Reference

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-nav__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    ChIP-seq Analysis Tutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Setup & Data
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Setup & Data
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_setup_environment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Environment Setup
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_bash_automation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Bash Automation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_geo_fastq_download/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GEO Fastq Download
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_fastq_concepts/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fastq Concepts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Alignment & QC
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Alignment & QC
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05_alignment_bowtie2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Alignment with Bowtie2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06_duplicate_removal_qc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Duplicate Removal & QC
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_library_complexity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Library Complexity
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../08_bam_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Quality Metrics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../09_strand_cross_correlation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Strand Cross Correlation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../10_bam_summary_fingerprint/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Summary & Fingerprint
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Peak Calling
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Peak Calling
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../11_macs3_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MACS3 Peak Calling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../12_frip_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Metrics - FRiP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility Analysis
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility Analysis
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-reproducibility-analysis-idr-irreproducible-discovery-rate" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Reproducibility Analysis: IDR (Irreproducible Discovery Rate)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Reproducibility Analysis: IDR (Irreproducible Discovery Rate)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#why-idr-matters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why IDR Matters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installing-idr" class="md-nav__link">
    <span class="md-ellipsis">
      
        Installing IDR
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-running-idr-on-cebpa-replicates" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Running IDR on CEBPA Replicates
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Running IDR on CEBPA Replicates">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#interpreting-idr-output" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interpreting IDR Output
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-for-reproducible-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Filtering for Reproducible Peaks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-idr-diagnostic-plots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding IDR Diagnostic Plots
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-motif-analysis-finding-dna-binding-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Motif Analysis: Finding DNA Binding Sequences
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Motif Analysis: Finding DNA Binding Sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#why-motif-analysis-matters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why Motif Analysis Matters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prerequisites
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-motif-discovery-with-homer" class="md-nav__link">
    <span class="md-ellipsis">
      
        4. Motif Discovery with HOMER
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="4. Motif Discovery with HOMER">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#installing-homer" class="md-nav__link">
    <span class="md-ellipsis">
      
        Installing HOMER
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#running-homer-on-idr-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Running HOMER on IDR Peaks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-homer-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interpreting HOMER Results
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-directory-structure-after-idr-motif-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        5. Directory Structure After IDR &amp; Motif Analysis
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        6. Summary
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Visualization
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Visualization
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../14_bigwig_generation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BigWig Generation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../15_visualization_heatmaps/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Visualization Heatmaps
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../16_chipseeker_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Annotation with ChIPseeker
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial_book_creation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    How We Built This Book
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../CONTRIBUTORS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributors
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-reproducibility-analysis-idr-irreproducible-discovery-rate" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Reproducibility Analysis: IDR (Irreproducible Discovery Rate)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Reproducibility Analysis: IDR (Irreproducible Discovery Rate)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#why-idr-matters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why IDR Matters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#installing-idr" class="md-nav__link">
    <span class="md-ellipsis">
      
        Installing IDR
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-running-idr-on-cebpa-replicates" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Running IDR on CEBPA Replicates
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Running IDR on CEBPA Replicates">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#interpreting-idr-output" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interpreting IDR Output
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-for-reproducible-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Filtering for Reproducible Peaks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-idr-diagnostic-plots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding IDR Diagnostic Plots
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-motif-analysis-finding-dna-binding-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Motif Analysis: Finding DNA Binding Sequences
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Motif Analysis: Finding DNA Binding Sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#why-motif-analysis-matters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why Motif Analysis Matters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prerequisites
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-motif-discovery-with-homer" class="md-nav__link">
    <span class="md-ellipsis">
      
        4. Motif Discovery with HOMER
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="4. Motif Discovery with HOMER">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#installing-homer" class="md-nav__link">
    <span class="md-ellipsis">
      
        Installing HOMER
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#running-homer-on-idr-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Running HOMER on IDR Peaks
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-homer-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interpreting HOMER Results
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-directory-structure-after-idr-motif-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        5. Directory Structure After IDR &amp; Motif Analysis
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        6. Summary
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="reproducibility-analysis-idr-consensus-peaks-motif-discovery">Reproducibility Analysis - IDR, Consensus Peaks &amp; Motif Discovery<a class="headerlink" href="#reproducibility-analysis-idr-consensus-peaks-motif-discovery" title="Permanent link">&para;</a></h1>
<p><code>IDR</code> <code>reproducibility</code> <code>consensus-peaks</code> <code>motif-analysis</code> <code>HOMER</code> <code>MEME-ChIP</code> <code>ENCODE</code> <code>peak-filtering</code></p>
<hr />
<h2 id="1-reproducibility-analysis-idr-irreproducible-discovery-rate">1. Reproducibility Analysis: IDR (Irreproducible Discovery Rate)<a class="headerlink" href="#1-reproducibility-analysis-idr-irreproducible-discovery-rate" title="Permanent link">&para;</a></h2>
<p><strong>IDR</strong> (<a href="https://github.com/nboley/idr#output-file-format">https://github.com/nboley/idr#output-file-format</a>) is a statistical framework used to quantify how consistently peaks are detected between biological replicates. It was developed for large-scale functional genomics projects and is mandated by <strong>ENCODE</strong> for ChIP-seq reproducibility assessment.</p>
<p>IDR models peak rankings between replicates and estimates the probability that a peak is irreproducible noise rather than a real signal.</p>
<h3 id="why-idr-matters">Why IDR Matters<a class="headerlink" href="#why-idr-matters" title="Permanent link">&para;</a></h3>
<ul>
<li>Peaks unique to one replicate are likely noise</li>
<li>Peaks present in both replicates are high-confidence</li>
<li>IDR &lt; 0.05 means peaks are reproducible</li>
</ul>
<p>ENCODE describes <a href="https://www.encodeproject.org/data-standards/terms/">two ways</a> to define reproducible peaks using IDR. The conservative approach, used here, runs IDR on true biological replicates and keeps only peaks that agree between independent experiments, giving high-confidence results. The optimal approach runs IDR on pseudoreplicates created by randomly splitting pooled data, which increases sensitivity. In this tutorial, we use the conservative approach with the biological replicates ceb_ENCFF327JFG_peaks and ceb_ENCFF744SVA_peaks.narrowPeak to focus on reproducible peaks.</p>
<h3 id="installing-idr">Installing IDR<a class="headerlink" href="#installing-idr" title="Permanent link">&para;</a></h3>
<p><strong>Install IDR</strong> (if not already installed):</p>
<p>IDR has <strong>binary and dependency issues on macOS ARM64</strong>, and it is recommended to create a <strong>dedicated environment</strong> to avoid conflicts with other packages.</p>
<div class="highlight"><pre><span></span><code>name: idr_env
channels:
  - conda-forge
  - bioconda
dependencies:
  - python=3.9
  - numpy=1.23.5
  - scipy=1.10.1
  - matplotlib=3.7.1
  - pip
  - pip:
      - git+https://github.com/nboley/idr.git
</code></pre></div>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>idr_env.yml
conda<span class="w"> </span>activate<span class="w"> </span>idr_env
</code></pre></div>
<hr />
<h2 id="2-running-idr-on-cebpa-replicates">2. Running IDR on CEBPA Replicates<a class="headerlink" href="#2-running-idr-on-cebpa-replicates" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>idr<span class="w"> </span>--samples<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>macs3_results/ceb_ENCFF327JFG_peaks.narrowPeak<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>macs3_results/ceb_ENCFF744SVA_peaks.narrowPeak<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--input-file-type<span class="w"> </span>narrowPeak<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--rank<span class="w"> </span>signal.value<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--output-file<span class="w"> </span>idr/ceb_idr_peaks.txt<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--plot<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--log-output-file<span class="w"> </span>idr/ceb_idr.log
</code></pre></div>
<p><strong>Parameter Explanation:</strong></p>
<ul>
<li><code>--samples</code>: The two replicate peak files</li>
<li><code>--rank p.value</code>: Rank peaks by p-value (ENCODE default)</li>
<li>Alternatives: <code>signal.value</code> (fold enrichment) or <code>q.value</code> (FDR)</li>
<li><code>--plot</code>: Generate diagnostic plots showing replicate concordance</li>
<li><code>--log-output-file</code>: Save detailed statistics</li>
</ul>
<h3 id="interpreting-idr-output">Interpreting IDR Output<a class="headerlink" href="#interpreting-idr-output" title="Permanent link">&para;</a></h3>
<p>The output file <code>ceb_idr_peaks.txt</code> contains peaks found in both replicates with IDR scores. See the <a href="https://github.com/nboley/idr">official IDR documentation</a> for complete format details.</p>
<p><strong>Key Columns:</strong></p>
<p><strong>Columns 1-3 (chr, start, end):</strong> Genomic coordinates of the merged peak region where reproducible signal was detected.</p>
<p><strong>Column 5 (scaled IDR score):</strong> Compressed 0-1000 score for ranking and visualization only (NOT for filtering). Higher values indicate better reproducibility:</p>
<ul>
<li>IDR = 0 → score = 1000 (perfect)</li>
<li>IDR = 0.05 → score ≈ 540</li>
</ul>
<p><strong>Column 7 (signalValue):</strong> Strength of ChIP enrichment inherited from MACS3. Reflects signal intensity, not reproducibility.</p>
<p><strong>Column 11 (local IDR):</strong> Reproducibility score for this specific peak using <code>-log10(IDR)</code> scale.</p>
<p><strong>Column 12 (global IDR):</strong> <strong>Primary ENCODE metric</strong> for defining reproducible peaks across replicates, also on <code>-log10(IDR)</code> scale where higher values = better reproducibility:</p>
<ul>
<li><code>-log10(IDR) ≥ 1.3</code> → IDR ≤ 0.05 (standard threshold)</li>
<li><code>-log10(IDR) ≥ 2.0</code> → IDR ≤ 0.01 (very stringent)</li>
<li>Formula: <code>-log10(0.05) = 1.301 ≈ 1.3</code></li>
</ul>
<p><strong>Check peak counts at different thresholds:</strong></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Standard threshold (IDR ≤ 0.05)</span>
awk<span class="w"> </span><span class="s1">&#39;$12 &gt;= 1.3&#39;</span><span class="w"> </span>idr/ceb_idr_peaks.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l

<span class="c1"># Stringent threshold (IDR ≤ 0.01)</span>
awk<span class="w"> </span><span class="s1">&#39;$12 &gt;= 2.0&#39;</span><span class="w"> </span>idr/ceb_idr_peaks.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>wc<span class="w"> </span>-l
</code></pre></div>
<hr />
<h3 id="filtering-for-reproducible-peaks">Filtering for Reproducible Peaks<a class="headerlink" href="#filtering-for-reproducible-peaks" title="Permanent link">&para;</a></h3>
<p>Filter peaks that pass the standard IDR threshold:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$12 &gt;= 1.3&#39;</span><span class="w"> </span>idr/ceb_idr_peaks.txt<span class="w"> </span>&gt;<span class="w"> </span>idr/ceb_idr_passed.bed
</code></pre></div>
<div class="highlight"><pre><span></span><code># Count results

(chip) rajaishaqnabikhan@Rajas-MacBook-Pro bws % wc -l &lt;  idr/ceb_idr_peaks.txt
    32273
(chip) rajaishaqnabikhan@Rajas-MacBook-Pro bws % wc -l &lt;  idr/ceb_idr_passed.bed
     9468
</code></pre></div>
<p><strong>Interpretation:</strong></p>
<p>Roughly 29.3% (9,468/32,273) of initial peaks are reproducible at IDR ≤ 0.05. This is entirely reasonable for a transcription factor ChIP-seq, where we expect only the strongest, most reproducible binding sites to pass stringent filtering.</p>
<hr />
<h3 id="understanding-idr-diagnostic-plots">Understanding IDR Diagnostic Plots<a class="headerlink" href="#understanding-idr-diagnostic-plots" title="Permanent link">&para;</a></h3>
<p>IDR generates three diagnostic plots to visualize replicate concordance:</p>
<p><strong>Plot 1: Rank Consistency Plot</strong></p>
<p><img alt="image" src="../images/idr_rank_plot.jpg" /></p>
<p>This plot displays the normalized rank of peaks in replicate 1 (X-axis) versus replicate 2 (Y-axis). Each point represents a peak shared between replicates, with black/gray points indicating reproducible peaks (IDR &lt; 0.05) and red points showing irreproducible peaks (IDR ≥ 0.05). A strong diagonal indicates good replicate concordance, where high-ranking peaks in one replicate are also high-ranking in the other.</p>
<p><strong>Plot 2: Signal Consistency Plot</strong></p>
<p><img alt="image" src="../images/idr_signal_plot.jpg" /></p>
<p>This log10(signal)-log10(signal) plot compares signal strength between replicates, with black points representing reproducible peaks (IDR &lt; 0.05) and red points showing irreproducible peaks (IDR ≥ 0.05). A tight diagonal band for black points confirms that strong peaks are consistently strong in both replicates, validating the biological signal.</p>
<p><strong>Plot 3: IDR vs Peak Rank</strong></p>
<p><img alt="image" src="../images/idr_score_plot.jpg" /></p>
<p>This plot shows peak rank bins for replicate 1 (X-axis, with best peaks on the right) versus -log10(IDR) scores (Y-axis). Individual points represent peaks, with boxplots summarizing IDR distributions within rank bins. The dashed horizontal line marks the IDR = 0.05 cutoff (-log10 ≈ 1.3). A sigmoidal curve where -log10(IDR) increases sharply as peak rank improves indicates that top-ranked peaks have high reproducibility scores, which is expected for quality ChIP-seq data.</p>
<hr />
<p><strong>Example of Poor IDR (Cross-Factor Comparison):</strong></p>
<p>This demonstrates poor IDR results when calculated between biologically unrelated samples (CEBPA replicate vs H3K9ac replicate), shown here for illustration purposes only:</p>
<p><img alt="H3K9ac_ceb_idr_peaks txt" src="../images/idr_cross_factor.png" /></p>
<h2 id="3-motif-analysis-finding-dna-binding-sequences">3. Motif Analysis: Finding DNA Binding Sequences<a class="headerlink" href="#3-motif-analysis-finding-dna-binding-sequences" title="Permanent link">&para;</a></h2>
<h3 id="why-motif-analysis-matters">Why Motif Analysis Matters<a class="headerlink" href="#why-motif-analysis-matters" title="Permanent link">&para;</a></h3>
<p>Reproducible peaks answer <em>where</em> the protein binds.
Motif analysis answers <em>why</em> it binds there.</p>
<ul>
<li>Transcription factors bind <strong>specific DNA sequences</strong></li>
<li>Histone marks do <strong>not</strong> bind DNA directly</li>
<li>Motif enrichment validates biological specificity</li>
</ul>
<p>Example:</p>
<ul>
<li><strong>CEBPA:</strong> sequence-specific TF</li>
<li><strong>H3K9ac:</strong> chromatin mark, no intrinsic motif</li>
</ul>
<p>If a TF ChIP-seq lacks its expected motif, the experiment is suspect.</p>
<h3 id="prerequisites">Prerequisites<a class="headerlink" href="#prerequisites" title="Permanent link">&para;</a></h3>
<p>You need the <strong>genome FASTA file</strong> for sequence extraction:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Download and extract genome FASTA</span>
wget<span class="w"> </span>https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_49/GRCh38.primary_assembly.genome.fa.gz
gunzip<span class="w"> </span>GRCh38.primary_assembly.genome.fa.gz
</code></pre></div>
<h2 id="4-motif-discovery-with-homer">4. Motif Discovery with HOMER<a class="headerlink" href="#4-motif-discovery-with-homer" title="Permanent link">&para;</a></h2>
<p><strong>HOMER (Hypergeometric Optimization of Motif EnRichment)</strong> is the most widely used ChIP-seq motif discovery framework.</p>
<h3 id="installing-homer">Installing HOMER<a class="headerlink" href="#installing-homer" title="Permanent link">&para;</a></h3>
<blockquote>
<p>[!NOTE]
<strong>macOS ARM64 Users:</strong> HOMER may require manual installation. Use the provided <code>install_homer.sh</code> script for streamlined setup.</p>
</blockquote>
<hr />
<h3 id="running-homer-on-idr-peaks">Running HOMER on IDR Peaks<a class="headerlink" href="#running-homer-on-idr-peaks" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1"># Extract genomic coordinates from IDR peaks</span>
awk<span class="w"> </span><span class="s1">&#39;$12 &gt;= 1.3 {print $1,$2,$3}&#39;</span><span class="w"> </span><span class="nv">OFS</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>idr/ceb_idr_peaks.txt<span class="w"> </span>&gt;<span class="w"> </span>idr/ceb_idr_passed.bed

<span class="c1"># Run HOMER motif discovery</span>
findMotifsGenome.pl<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>idr/ceb_idr_passed.bed<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>GRCh38.primary_assembly.genome.fa<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>idr/cebpa_motifs/<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-size<span class="w"> </span><span class="m">200</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-mask<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-p<span class="w"> </span><span class="m">8</span>
</code></pre></div>
<p><strong>Parameter Explanation:</strong></p>
<ul>
<li><code>idr/ceb_idr_passed.bed</code>: Input peak file (IDR-filtered peaks)</li>
<li><code>GRCh38.primary_assembly.genome.fa</code>: Genome assembly</li>
<li><code>idr/cebpa_motifs/</code>: Output directory for results</li>
<li><code>-size 200</code>: Search ±100bp around peak center (200bp total window)</li>
<li><code>-mask</code>: Mask repetitive DNA sequences</li>
<li><code>-p 8</code>: Use 8 CPU threads for parallel processing</li>
</ul>
<h3 id="interpreting-homer-results">Interpreting HOMER Results<a class="headerlink" href="#interpreting-homer-results" title="Permanent link">&para;</a></h3>
<p>HOMER creates an output directory with:</p>
<div class="highlight"><pre><span></span><code>idr/cebpa_motifs/
├── homerResults.html          # Main results page (OPEN THIS)
├── knownResults.html          # Matches to known motifs
├── motif1.logo.png            # Top de novo motif
├── motif2.logo.png
└── motif1.motif               # Position weight matrix
</code></pre></div>
<p><strong>Example output from CEBPA analysis:</strong></p>
<p><img alt="Screenshot 2025-12-22 at 2 31 56 PM" src="../images/homer_results.png" /></p>
<p><strong>Interpreting the results:</strong></p>
<ul>
<li><strong>Rank (1):</strong> Top-ranked motif, most strongly enriched among all detected sequences</li>
<li><strong>Motif (logo):</strong> Canonical C/EBP-family DNA-binding motif (TTGCGCAA/TTGCAT core) indicating specific TF binding</li>
<li><strong>P-value (1e−2876):</strong> Extreme overrepresentation in peaks vs background, statistically highly significant</li>
<li><strong>% of Targets (63.39%):</strong> Nearly two-thirds of peaks contain this motif, indicating coherent TF-driven peak set</li>
<li><strong>% of Background (3.43%):</strong> Rare in background regions, demonstrating strong specificity and enrichment</li>
<li><strong>STD (30.7 bp vs 64.0 bp):</strong> Motif centered near peak summits (targets) vs broadly distributed (background), supporting direct DNA binding</li>
<li><strong>Best Match (CEBPE/MA0837.1, 0.963):</strong> De novo motif matches known JASPAR CEBPE motif with high confidence</li>
<li><strong>Motif File:</strong> Provides position weight matrix for reproducibility and downstream scanning</li>
</ul>
<hr />
<h2 id="5-directory-structure-after-idr-motif-analysis">5. Directory Structure After IDR &amp; Motif Analysis<a class="headerlink" href="#5-directory-structure-after-idr-motif-analysis" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
├── macs3_results/              ← Peak files from MACS3
│   ├── ceb_ENCFF327JFG_peaks.narrowPeak
│   ├── ceb_ENCFF744SVA_peaks.narrowPeak
│   └── ...
├── idr/                        ← IDR outputs
│   ├── ceb_idr_peaks.txt       # All peaks with IDR scores
│   ├── ceb_idr_passed.bed      # Filtered peaks (IDR ≤ 0.05)
│   ├── ceb_idr.log             # IDR statistics
│   └── ceb_idr_peaks.txt.png   # Diagnostic plots
└── idr/cebpa_motifs/           ← HOMER motif results
    ├── homerResults.html       # Main results page
    ├── knownResults.html       # Known motif matches
    ├── motif1.logo.png         # Top de novo motif
    └── motif1.motif            # Position weight matrix
</code></pre></div>
<hr />
<h2 id="6-summary">6. Summary<a class="headerlink" href="#6-summary" title="Permanent link">&para;</a></h2>
<p><strong>Key Achievements:</strong></p>
<ol>
<li>Filtered 32,273 peaks → 9,468 reproducible peaks (29.3%) using IDR ≤ 0.05</li>
<li>Validated replicate concordance with diagnostic plots</li>
<li>Identified canonical C/EBP motif in 63% of peaks (HOMER)</li>
<li>Confirmed CEBPA specificity through motif enrichment</li>
</ol>
<p><strong>Quality Indicators:</strong></p>
<p><strong>Good:</strong> Strong diagonal in rank-rank plot, ~30% IDR pass rate, expected motif in &gt;50% peaks
<strong>Poor:</strong> Scattered plot, &lt;10% IDR pass rate, no motif enrichment</p>
<blockquote>
<p>[!NOTE]
<strong>Up Next:</strong> Generate normalized signal tracks (BigWig files) for genome browser visualization and create publication-quality heatmaps showing enrichment patterns around genomic features.</p>
</blockquote>
<hr />












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 Raja Ishaq Nabi Khan
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/ishaaq34" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.top", "search.suggest", "search.highlight", "content.code.copy"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>