
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A comprehensive, hands-on tutorial for ChIP-seq data analysis">
      
      
        <meta name="author" content="Raja Ishaq Nabi Khan">
      
      
        <link rel="canonical" href="https://ishaaq34.github.io/Chipseq_analysis_tutorial_mkdocs/06_duplicate_removal_qc/">
      
      
        <link rel="prev" href="../05_alignment_bowtie2/">
      
      
        <link rel="next" href="../07_library_complexity/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>Duplicate Removal & QC - ChIP-seq Analysis Tutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="light-blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#handling-duplicates-quality-control" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-header__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq Analysis Tutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Duplicate Removal & QC
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="light-blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="light-blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../00_introduction/" class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../01_setup_environment/" class="md-tabs__link">
          
  
  
  Setup & Data

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../05_alignment_bowtie2/" class="md-tabs__link">
          
  
  
  Alignment & QC

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../11_macs3_peak_calling/" class="md-tabs__link">
          
  
  
  Peak Calling

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../14_bigwig_generation/" class="md-tabs__link">
          
  
  
  Visualization

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../tutorial_book_creation.md" class="md-tabs__link">
          
  
  
  Reference

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-nav__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    ChIP-seq Analysis Tutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Setup & Data
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Setup & Data
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_setup_environment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Environment Setup
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_bash_automation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Bash Automation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_geo_fastq_download/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GEO Fastq Download
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_fastq_concepts/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fastq Concepts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Alignment & QC
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Alignment & QC
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05_alignment_bowtie2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Alignment with Bowtie2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Duplicate Removal & QC
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Duplicate Removal & QC
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-basic-concept-the-photocopier-analogy" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Basic Concept: The "Photocopier" Analogy
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-understanding-the-details" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Understanding the Details
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Understanding the Details">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pcr-vs-optical-duplicates" class="md-nav__link">
    <span class="md-ellipsis">
      
        PCR vs. Optical Duplicates
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-role-of-read-groups-rg" class="md-nav__link">
    <span class="md-ellipsis">
      
        The Role of Read Groups (RG)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-marking-removing-duplicates-why-picard-is-unique" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Marking &amp; Removing Duplicates: Why Picard is Unique
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_library_complexity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Library Complexity
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../08_bam_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Quality Metrics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../09_strand_cross_correlation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Strand Cross Correlation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../10_bam_summary_fingerprint/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Summary & Fingerprint
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Peak Calling
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Peak Calling
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../11_macs3_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MACS3 Peak Calling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../12_frip_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Metrics - FRiP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../13_idr_consensus_motifs_rk_corrected/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Visualization
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Visualization
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../14_bigwig_generation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BigWig Generation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../15_visualization_heatmaps/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Visualization Heatmaps
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../16_chipseeker_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Annotation with ChIPseeker
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial_book_creation.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    How We Built This Book
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../CONTRIBUTORS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributors
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-basic-concept-the-photocopier-analogy" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Basic Concept: The "Photocopier" Analogy
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-understanding-the-details" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Understanding the Details
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Understanding the Details">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pcr-vs-optical-duplicates" class="md-nav__link">
    <span class="md-ellipsis">
      
        PCR vs. Optical Duplicates
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-role-of-read-groups-rg" class="md-nav__link">
    <span class="md-ellipsis">
      
        The Role of Read Groups (RG)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-marking-removing-duplicates-why-picard-is-unique" class="md-nav__link">
    <span class="md-ellipsis">
      
        3. Marking &amp; Removing Duplicates: Why Picard is Unique
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="handling-duplicates-quality-control">Handling Duplicates &amp; Quality Control<a class="headerlink" href="#handling-duplicates-quality-control" title="Permanent link">&para;</a></h1>
<p><code>PCR-duplicates</code> <code>optical-duplicates</code> <code>Picard</code> <code>samtools</code> <code>MarkDuplicates</code> <code>deduplication</code> <code>read-groups</code> <code>MultiQC</code></p>
<h2 id="1-basic-concept-the-photocopier-analogy">1. Basic Concept: The "Photocopier" Analogy<a class="headerlink" href="#1-basic-concept-the-photocopier-analogy" title="Permanent link">&para;</a></h2>
<p>Imagine you are trying to read a rare, handwritten manuscript (your DNA sample). You want to digitize it, so you take photos (sequencing reads) of different pages.</p>
<ul>
<li><strong>Real Signal (Enriched Regions):</strong> If many people are taking photos of the <em>same important page</em> because it's interesting, that's good! In ChIP-seq, this happens when a protein binds strongly to a specific DNA spot. We see many reads there because the biological signal is strong.</li>
<li><strong>Duplicates (Artifacts):</strong> Now, imagine the photocopier gets stuck and prints 100 copies of a <em>random, unimportant page</em> just because of a machine error. These copies don't mean that page is 100 times more important; they are just <strong>junk</strong>. In sequencing, this is called <strong>PCR duplication</strong>—where the chemistry accidentally over-copies a single DNA fragment.</li>
</ul>
<p><strong>Goal:</strong> We want to keep the "popular pages" (real biological signal) but throw away the "accidental machine copies" (PCR duplicates) so they don't trick us into thinking a random spot is important.</p>
<blockquote>
<p>[!NOTE]
<strong>Two Tools, Same Goal:</strong> This chapter covers two methods for duplicate removal. <strong>Picard</strong> (Section 3) offers fine-grained control with read groups and optical duplicate detection—ideal for complex multi-replicate studies. <strong>Samtools</strong> (Section 4) provides a simpler, single-pipeline approach—sufficient for most ChIP-seq analyses. Choose the method that fits your workflow.</p>
</blockquote>
<hr />
<h2 id="2-understanding-the-details">2. Understanding the Details<a class="headerlink" href="#2-understanding-the-details" title="Permanent link">&para;</a></h2>
<p>For those who want to understand the "under the hood" mechanics, here is what are different types of duplicates and  the Read Groups .</p>
<h3 id="pcr-vs-optical-duplicates">PCR vs. Optical Duplicates<a class="headerlink" href="#pcr-vs-optical-duplicates" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Type</th>
<th>Origin</th>
<th>Typical Cause</th>
<th>Implication</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>PCR Duplicates</strong></td>
<td>Library Prep</td>
<td>Over-cycling (amplifying DNA too much)</td>
<td>Shows low library complexity (not enough unique DNA to start with).</td>
</tr>
<tr>
<td><strong>Optical Duplicates</strong></td>
<td>Sequencer</td>
<td>Camera errors reads the same cluster as two</td>
<td>Technical glitch on the flow cell.</td>
</tr>
</tbody>
</table>
<h3 id="the-role-of-read-groups-rg">The Role of Read Groups (RG)<a class="headerlink" href="#the-role-of-read-groups-rg" title="Permanent link">&para;</a></h3>
<p>A <strong>Read Group</strong> is a tag that tells the software "this read came from Sample A, Run 1."</p>
<ul>
<li><strong>Why is it vital?</strong> If you merge two different samples (e.g., Replicate 1 and Replicate 2), you might have two different reads that coincidentally map to the same spot.</li>
<li>Without Read Groups, Picard acts blindly: "These look identical! Delete one!" -&gt; <strong>Data Loss.</strong></li>
<li>With Read Groups, Picard sees: "Oh, one is from Replicate 1 and one is from Replicate 2. They are different samples. Keep both!"</li>
</ul>
<p><strong>*Full hierarchy in RGing</strong></p>
<div class="highlight"><pre><span></span><code>RGSM  → biological sample (Control H3k9ac)
  └── RGLB → library prep (usually lib1 however if Different library preps (even from same sample) RGLB=lib1 RGLB=lib2 )
        └── RGPU → flowcell + lane (from header of Fastq file )
              └── RGID → unique ID tying it all together (replicates: H3K27me3_IP_rep1, H3K27me3_IP_rep2)
</code></pre></div>
<h2 id="3-marking-removing-duplicates-why-picard-is-unique">3. Marking &amp; Removing Duplicates: Why Picard is Unique<a class="headerlink" href="#3-marking-removing-duplicates-why-picard-is-unique" title="Permanent link">&para;</a></h2>
<p><strong>Picard</strong> stands out among duplicate-marking tools because it:</p>
<ul>
<li>Distinguishes <strong>optical vs. PCR duplicates</strong> during QC reporting.  </li>
<li>Leverages <strong>read group (RG) identifiers</strong> to avoid over-collapsing reads when combining multiple replicates or lanes.  </li>
<li>Provides metrics for each library or sample, enabling fine-grained quality control.</li>
</ul>
<p>Picard’s RG-aware logic ensures duplicates are flagged <strong>within</strong>, but not <strong>across</strong>, biological replicates or lanes — preventing false duplicate marking when BAMs are merged.</p>
<hr />
<h1 id="31-adding-rgs-to-each-bam-file">3.1 Adding RGs to each Bam file<a class="headerlink" href="#31-adding-rgs-to-each-bam-file" title="Permanent link">&para;</a></h1>
<p><strong>Minimal augmented RG setup (merge BAM replicates)</strong></p>
<p>This read-group configuration is designed for the simplest defensible case: merging biological or technical ChIP-seq replicates and proceeding directly to peak calling.</p>
<p><strong>Input files needed:</strong></p>
<ul>
<li>Filtered BAM file: <code>bowalign_filtered/H3K27me3_IP_rep1.filtered.bam</code> (from section 05)</li>
</ul>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>picard_rg_bam

picard<span class="w"> </span>AddOrReplaceReadGroups<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">I</span><span class="o">=</span>bowalign_filtered/H3K27me3_IP_rep1.filtered.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">O</span><span class="o">=</span>picard_rg_bam/H3K27me3_IP_rep1.RG.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">RGID</span><span class="o">=</span>H3K27me3_IP_rep1<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">RGSM</span><span class="o">=</span>H3K9ac
</code></pre></div>
<p><strong>What this does:</strong></p>
<ul>
<li><strong>I=</strong> specifies the input BAM file from the alignment step</li>
<li><strong>O=</strong> specifies the output BAM file with read-group tags added</li>
<li><strong>RGID=</strong> sets a unique Read Group ID (one per replicate)</li>
<li><strong>RGSM=</strong> sets the biological sample name (used for grouping replicates)</li>
</ul>
<p><strong>Minimal augmented RG setup (optical duplicates enabled)</strong></p>
<p>This configuration extends the previous one by adding the minimum required metadata to make optical duplicate detection meaningful.
The addition of RGPU, encoding the flowcell and lane, defines the physical neighborhood in which optical duplicates can occur.</p>
<div class="highlight"><pre><span></span><code>picard<span class="w"> </span>AddOrReplaceReadGroups<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">I</span><span class="o">=</span>bowalign_filtered/H3K27me3_IP_rep1.filtered.bam<span class="w"> </span><span class="se">\ </span><span class="w">    </span><span class="c1"># Input: MAPQ-filtered BAM from section 05</span>
<span class="w">  </span><span class="nv">O</span><span class="o">=</span>picard_rg_bam/H3K27me3_IP_rep1.RG.bam<span class="w"> </span><span class="se">\ </span><span class="w">          </span><span class="c1"># Output with RG tags</span>
<span class="w">  </span><span class="nv">RGID</span><span class="o">=</span>H3K27me3_IP_rep1<span class="w"> </span><span class="se">\ </span><span class="w">                     </span><span class="c1"># Read Group ID</span>
<span class="w">  </span><span class="nv">RGSM</span><span class="o">=</span>H3K9ac<span class="w"> </span><span class="se">\ </span><span class="w">                         </span><span class="c1"># Biological sample</span>
<span class="w">  </span><span class="nv">RGPL</span><span class="o">=</span>ILLUMINA<span class="w"> </span><span class="se">\ </span><span class="w">                       </span><span class="c1"># REQUIRED for optical duplicate logic</span>
<span class="w">  </span><span class="nv">RGPU</span><span class="o">=</span>CA0TUACXX.1<span class="w">                       </span><span class="c1"># REQUIRED: flowcell.lane (from FASTQ header)</span>
</code></pre></div>
<p>More from <a href="https://broadinstitute.github.io/picard/command-line-overview.html#AddOrReplaceReadGroups">GATK: Read Groups</a>
           <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard">Picard markduplicates</a></p>
<h3 id="step-32-mark-duplicates-be-careful">Step 3.2: Mark Duplicates (Be Careful!)<a class="headerlink" href="#step-32-mark-duplicates-be-careful" title="Permanent link">&para;</a></h3>
<p>First, we will just <strong>mark</strong> the duplicates but <strong>keep them</strong> in the file. This is like highlighting the duplicate pages in yellow but not throwing them in the trash yet. This is important for Quality Control (QC) to see how bad the duplication problem is.</p>
<p><strong>Input files needed:</strong></p>
<ul>
<li>Filtered BAM file with read groups: <code>picard_rg_bam/H3K27me3_IP_rep1.RG.bam</code></li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># ----- 3.3 Mark duplicates (keep all reads, just mark) -----</span>

mkdir<span class="w"> </span>-p<span class="w"> </span>picard_markdup<span class="w"> </span>picard_markdup_metrics

<span class="c1"># Run Picard MarkDuplicates</span>
picard<span class="w"> </span>MarkDuplicates<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">I</span><span class="o">=</span>picard_rg_bam/H3K27me3_IP_rep1.RG.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">O</span><span class="o">=</span>picard_markdup/H3K27me3_IP_rep1.marked.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">M</span><span class="o">=</span>picard_markdup_metrics/H3K27me3_IP_rep1.metrics.txt<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">REMOVE_DUPLICATES</span><span class="o">=</span><span class="nb">false</span>

<span class="c1"># Index the new file</span>
samtools<span class="w"> </span>index<span class="w"> </span>picard_markdup/H3K27me3_IP_rep1.marked.bam
</code></pre></div>
<p><strong>What this does:</strong></p>
<ul>
<li><strong>I=</strong> specifies input BAM file with read groups</li>
<li><strong>O=</strong> specifies output BAM file with duplicates marked (not removed)</li>
<li><strong>M=</strong> creates metrics file reporting duplicate statistics</li>
<li><strong>REMOVE_DUPLICATES=false</strong> marks duplicates but keeps all reads in the file</li>
<li><strong>samtools index</strong> creates index for marked BAM file</li>
</ul>
<blockquote>
<p>[!IMPORTANT]
<strong>Why keep marked files?</strong> The Picard metrics file (<code>.metrics.txt</code>) contains duplicate rates that <strong>MultiQC</strong> can aggregate into summary reports. However, for <strong>downstream analysis (peak calling, BigWig generation)</strong>, always use the <strong>deduplicated BAMs</strong> to prevent PCR artifacts from inflating your signal.</p>
<p><em>Note:</em> While MACS can handle duplicates internally via <code>--keep-dup</code> options, using pre-deduplicated BAMs is the recommended practice—it gives you explicit control and avoids relying on MACS's position-based duplicate detection, which differs from Picard/samtools fragment-level detection. [<a href="https://github.com/macs3-project/MACS/issues/33">GitHub #33</a>] [<a href="https://www.biostars.org/p/318974/#319028">Biostars</a>]</p>
</blockquote>
<p><strong>Aggregate metrics with MultiQC:</strong></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Run MultiQC on Picard metrics folder</span>
multiqc<span class="w"> </span>picard_dedup_metrics/<span class="w"> </span>-o<span class="w"> </span>multiqc_report/
</code></pre></div>
<p>This generates an interactive HTML report showing duplicate rates across all samples.</p>
<h3 id="step-33-remove-duplicates-clean-up">Step 3.3: Remove Duplicates (Clean Up)<a class="headerlink" href="#step-33-remove-duplicates-clean-up" title="Permanent link">&para;</a></h3>
<p>Now that we've checked the quality, we create a "clean" version of our data for analysis. We remove the marked duplicates so they don't interfere with peak calling.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># ----- 3.4 Create a duplicate-removed BAM by filtering marked BAM -----</span>

mkdir<span class="w"> </span>-p<span class="w"> </span>picard_dedup_bam<span class="w"> </span>picard_dedup_metrics

<span class="c1"># Run Picard to REMOVE duplicates</span>
picard<span class="w"> </span>MarkDuplicates<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">I</span><span class="o">=</span>picard_rg_bam/H3K27me3_IP_rep1.RG.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">O</span><span class="o">=</span>picard_dedup_bam/H3K27me3_IP_rep1.dedup.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">M</span><span class="o">=</span>picard_dedup_metrics/H3K27me3_IP_rep1.metrics.txt<span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="nv">REMOVE_DUPLICATES</span><span class="o">=</span><span class="nb">true</span>

<span class="c1"># Index the new file</span>
samtools<span class="w"> </span>index<span class="w"> </span>picard_dedup_bam/H3K27me3_IP_rep1.dedup.bam
</code></pre></div>
<ul>
<li><code>REMOVE_DUPLICATES=true</code>: This time, we actually delete the highlighted duplicates.</li>
</ul>
<h3 id="after-picard-deduplication">After Picard Deduplication<a class="headerlink" href="#after-picard-deduplication" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
├── fastq_raw/
├── fastq_cleaned/
├── genome_index/
├── bowalign/                    ← Aligned BAMs
├── bowalign_filtered/           ← Filtered BAMs
├── picard_rg_bam/               ← BAMs with Read Groups
│   ├── H3K27me3_IP_rep1.RG.bam
│   ├── H3K27me3_IP_rep1.RG.bam.bai
│   └── ...
├── picard_dedup_bam/            ← Final Deduplicated BAMs
│   ├── H3K27me3_IP_rep1.dedup.bam
│   ├── H3K27me3_IP_rep1.dedup.bam.bai
│   └── ...
├── picard_metrics/              ← Duplication Reports
│   ├── H3K27me3_IP_rep1.metrics.txt
│   └── ...
└── sample_id.txt
</code></pre></div>
<hr />
<h2 id="4-samtools-simple-alternative">4. Samtools (Simple Alternative)<a class="headerlink" href="#4-samtools-simple-alternative" title="Permanent link">&para;</a></h2>
<p>Samtools provides a lightweight and reliable way to handle duplicates without the complexity of read groups. Choose this option if you want a straightforward workflow.</p>
<h3 id="step-41-prepare-bam-for-duplicate-detection">Step 4.1: Prepare BAM for Duplicate Detection<a class="headerlink" href="#step-41-prepare-bam-for-duplicate-detection" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1"># Paired-end mates are placed next to each other. Groups reads by read name</span>
samtools<span class="w"> </span>collate<span class="w"> </span>-o<span class="w"> </span>temp/H3K27me3_IP_rep1.collate.bam<span class="w"> </span>bowalign_filtered/H3K27me3_IP_rep1.filtered.bam

<span class="c1"># Synchronize mate flags and add mate-related tags</span>
samtools<span class="w"> </span>fixmate<span class="w"> </span>-m<span class="w"> </span>temp/H3K27me3_IP_rep1.collate.bam<span class="w"> </span>temp/H3K27me3_IP_rep1.fixmate.bam

<span class="c1"># Coordinate sort for duplicate marking</span>
samtools<span class="w"> </span>sort<span class="w"> </span>-o<span class="w"> </span>temp/H3K27me3_IP_rep1.positionsort.bam<span class="w"> </span>temp/H3K27me3_IP_rep1.fixmate.bam
</code></pre></div>
<h3 id="step-42-mark-duplicates-keep-for-qc">Step 4.2: Mark Duplicates (Keep for QC)<a class="headerlink" href="#step-42-mark-duplicates-keep-for-qc" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1"># Mark duplicates but KEEP them in the output</span>
samtools<span class="w"> </span>markdup<span class="w"> </span>temp/H3K27me3_IP_rep1.positionsort.bam<span class="w"> </span>samtools_markdup/H3K27me3_IP_rep1.marked.bam

<span class="c1"># Index the file</span>
samtools<span class="w"> </span>index<span class="w"> </span>samtools_markdup/H3K27me3_IP_rep1.marked.bam
</code></pre></div>
<h3 id="step-43-remove-duplicates-clean-for-analysis">Step 4.3: Remove Duplicates (Clean for Analysis)<a class="headerlink" href="#step-43-remove-duplicates-clean-for-analysis" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1"># Remove duplicates with -r flag</span>
samtools<span class="w"> </span>markdup<span class="w"> </span>-r<span class="w"> </span>temp/H3K27me3_IP_rep1.positionsort.bam<span class="w"> </span>samtools_dedup_bam/H3K27me3_IP_rep1.dedup.bam

<span class="c1"># Index the file</span>
samtools<span class="w"> </span>index<span class="w"> </span>samtools_dedup_bam/H3K27me3_IP_rep1.dedup.bam
</code></pre></div>
<p>This samtools-based workflow is simpler than Picard and avoids the need for read-group metadata, while still providing accurate duplicate detection for paired-end data.</p>
<h3 id="step-44-automation-loop-all-samples">Step 4.4: Automation Loop (All Samples)<a class="headerlink" href="#step-44-automation-loop-all-samples" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="nb">set</span><span class="w"> </span>-euo<span class="w"> </span>pipefail

mkdir<span class="w"> </span>-p<span class="w"> </span>samtools_dedup_bam

<span class="k">while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>-r<span class="w"> </span>sample<span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Processing </span><span class="nv">$sample</span><span class="s2">...&quot;</span>

<span class="w">  </span><span class="c1"># All steps piped together - no temp files needed!</span>
<span class="w">  </span>samtools<span class="w"> </span>collate<span class="w"> </span>-u<span class="w"> </span>-O<span class="w"> </span><span class="s2">&quot;bowalign_filtered/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.filtered.bam&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>samtools<span class="w"> </span>fixmate<span class="w"> </span>-m<span class="w"> </span>-u<span class="w"> </span>-<span class="w"> </span>-<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>samtools<span class="w"> </span>sort<span class="w"> </span>-u<span class="w"> </span>-<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>samtools<span class="w"> </span>markdup<span class="w"> </span>-r<span class="w"> </span>-<span class="w"> </span><span class="s2">&quot;samtools_dedup_bam/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.dedup.bam&quot;</span>

<span class="w">  </span><span class="c1"># Index the final output</span>
<span class="w">  </span>samtools<span class="w"> </span>index<span class="w"> </span><span class="s2">&quot;samtools_dedup_bam/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.dedup.bam&quot;</span>

<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Finished </span><span class="nv">$sample</span><span class="s2">&quot;</span>
<span class="k">done</span><span class="w"> </span>&lt;<span class="w"> </span>sample_id.txt
</code></pre></div>
<p><strong>Why use pipes?</strong></p>
<div class="highlight"><pre><span></span><code>collate → fixmate → sort → markdup
  ↓         ↓        ↓       ↓
 pipe     pipe     pipe    file
</code></pre></div>
<ul>
<li><strong><code>-u</code> flag</strong>: Outputs uncompressed BAM for faster streaming between commands</li>
<li><strong><code>-</code> symbol</strong>: Represents stdin/stdout, allowing data to flow through the pipeline</li>
<li><strong>No temp files</strong>: Everything processes in memory, saving disk space and I/O time</li>
<li><strong>Faster</strong>: No waiting for disk writes between each step</li>
</ul>
<p>More <a href="https://www.htslib.org/algorithms/duplicate.html">samtools</a></p>
<h3 id="after-samtools-deduplication">After Samtools Deduplication<a class="headerlink" href="#after-samtools-deduplication" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
├── bowalign_filtered/           ← Input: MAPQ-filtered BAM files (from section 05)
│   ├── H3K27me3_IP_rep1.filtered.bam
│   └── ...
├── samtools_markdup/            ← Duplicates marked (not removed)
│   ├── H3K27me3_IP_rep1.marked.bam
│   └── ...
├── samtools_dedup_bam/          ← Duplicates removed
│   ├── H3K27me3_IP_rep1.dedup.bam
│   ├── H3K27me3_IP_rep1.dedup.bam.bai
│   └── ...
└── sample_id.txt
</code></pre></div>
<hr />
<h2 id="summary">Summary<a class="headerlink" href="#summary" title="Permanent link">&para;</a></h2>
<ol>
<li><strong>Understand:</strong> PCR duplicates inflate signal artificially; we mark/remove them.</li>
<li><strong>Action:</strong> Use Picard (with read groups) or samtools to handle duplicates.</li>
<li><strong>Result:</strong> Clean, deduplicated BAM files ready for peak calling.</li>
</ol>
<blockquote>
<p>[!NOTE]
<strong>Up Next:</strong> We'll assess library complexity to understand how deeply our libraries were sequenced.</p>
</blockquote>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 Raja Ishaq Nabi Khan
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/ishaaq34" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.top", "search.suggest", "search.highlight", "content.code.copy"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>