
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A comprehensive, hands-on tutorial for ChIP-seq data analysis">
      
      
        <meta name="author" content="Raja Ishaq Nabi Khan">
      
      
        <link rel="canonical" href="https://ishaaq34.github.io/Chipseq_analysis_tutorial_mkdocs/10_bam_summary_fingerprint/">
      
      
        <link rel="prev" href="../09_strand_cross_correlation/">
      
      
        <link rel="next" href="../11_macs3_peak_calling/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>BAM Summary & Fingerprint - ChIP-seq Analysis Tutorial</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="light-blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#advanced-qc-with-deeptools-the-health-checkup" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-header__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq Analysis Tutorial
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              BAM Summary & Fingerprint
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="light-blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="light-blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../00_introduction/" class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../01_setup_environment/" class="md-tabs__link">
          
  
  
  Setup & Data

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../05_alignment_bowtie2/" class="md-tabs__link">
          
  
  
  Alignment & QC

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../11_macs3_peak_calling/" class="md-tabs__link">
          
  
  
  Peak Calling

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../14_bigwig_generation/" class="md-tabs__link">
          
  
  
  Visualization

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../tutorial_book_creation.md" class="md-tabs__link">
          
  
  
  Reference

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq Analysis Tutorial" class="md-nav__button md-logo" aria-label="ChIP-seq Analysis Tutorial" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    ChIP-seq Analysis Tutorial
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ishaaq34/Chipseq_analysis_tutorial_mkdocs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    ishaaq34/Chipseq_analysis_tutorial_mkdocs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Setup & Data
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Setup & Data
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_setup_environment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Environment Setup
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_bash_automation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Bash Automation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_geo_fastq_download/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GEO Fastq Download
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_fastq_concepts/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fastq Concepts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Alignment & QC
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Alignment & QC
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05_alignment_bowtie2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Alignment with Bowtie2
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06_duplicate_removal_qc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Duplicate Removal & QC
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_library_complexity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Library Complexity
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../08_bam_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Quality Metrics
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../09_strand_cross_correlation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Strand Cross Correlation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    BAM Summary & Fingerprint
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    BAM Summary & Fingerprint
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-basic-concept-the-health-check" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Basic Concept (The Health Check)
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-running-the-qc-of-bam-files-before-peak-calling" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Running the QC of bam files before Peak Calling
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Running the QC of bam files before Peak Calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-fingerprint-plot" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.1 Fingerprint Plot
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-coverage-plot" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.2 Coverage Plot
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23-summary-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.3 Summary Matrix
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#24-correlation-pca" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.4 Correlation &amp; PCA
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#level-3-reading-the-charts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Level 3: Reading the Charts
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Level 3: Reading the Charts">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-interpreting-the-fingerprint-the-census" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.1 Interpreting the Fingerprint (The Census)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32-interpreting-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.2 Interpreting Coverage
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-interpreting-correlation-the-family-tree" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.3 Interpreting Correlation (The Family Tree)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-interpreting-pca" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.4 Interpreting PCA
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#directory-structure-after-deeptools-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directory Structure After deepTools QC
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        Summary
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Peak Calling
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Peak Calling
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../11_macs3_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MACS3 Peak Calling
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../12_frip_quality_metrics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Metrics - FRiP
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../13_idr_consensus_motifs_rk_corrected/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Visualization
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Visualization
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../14_bigwig_generation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BigWig Generation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../15_visualization_heatmaps/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Visualization Heatmaps
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../16_chipseeker_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Annotation with ChIPseeker
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial_book_creation.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    How We Built This Book
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../CONTRIBUTORS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributors
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-basic-concept-the-health-check" class="md-nav__link">
    <span class="md-ellipsis">
      
        1. Basic Concept (The Health Check)
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-running-the-qc-of-bam-files-before-peak-calling" class="md-nav__link">
    <span class="md-ellipsis">
      
        2. Running the QC of bam files before Peak Calling
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Running the QC of bam files before Peak Calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-fingerprint-plot" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.1 Fingerprint Plot
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-coverage-plot" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.2 Coverage Plot
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23-summary-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.3 Summary Matrix
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#24-correlation-pca" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.4 Correlation &amp; PCA
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#level-3-reading-the-charts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Level 3: Reading the Charts
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Level 3: Reading the Charts">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-interpreting-the-fingerprint-the-census" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.1 Interpreting the Fingerprint (The Census)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32-interpreting-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.2 Interpreting Coverage
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-interpreting-correlation-the-family-tree" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.3 Interpreting Correlation (The Family Tree)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-interpreting-pca" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.4 Interpreting PCA
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#directory-structure-after-deeptools-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Directory Structure After deepTools QC
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    <span class="md-ellipsis">
      
        Summary
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="advanced-qc-with-deeptools-the-health-checkup">Advanced QC with deepTools (The "Health Checkup")<a class="headerlink" href="#advanced-qc-with-deeptools-the-health-checkup" title="Permanent link">&para;</a></h1>
<p><code>deepTools</code> <code>plotFingerprint</code> <code>plotCoverage</code> <code>multiBamSummary</code> <code>correlation</code> <code>PCA</code> <code>ChIP-seq</code> <code>quality-control</code> <code>enrichment</code> <code>BAM-QC</code></p>
<h2 id="1-basic-concept-the-health-check">1. Basic Concept (The Health Check)<a class="headerlink" href="#1-basic-concept-the-health-check" title="Permanent link">&para;</a></h2>
<p>Before we call peaks, we must perform a <strong>Health Checkup</strong> on our data.</p>
<ul>
<li><strong>The Census (Fingerprint):</strong> Are the reads spread out evenly (Socialist/Input) or concentrated in specific spots (Capitalist/ChIP)?</li>
<li><strong>The Coverage Check:</strong> Do we have enough reads? Are they duplicates?</li>
<li><strong>The Family Tree (Correlation &amp; PCA):</strong> Do biological replicates (siblings) look similar? Are they distinct from the control?</li>
</ul>
<p>We use a suite of tools called <strong>deepTools</strong> to generate these reports.</p>
<hr />
<h2 id="2-running-the-qc-of-bam-files-before-peak-calling">2. Running the QC of bam files before Peak Calling<a class="headerlink" href="#2-running-the-qc-of-bam-files-before-peak-calling" title="Permanent link">&para;</a></h2>
<blockquote>
<p>[!IMPORTANT]
<strong>About ENCODE BAM files:</strong> The ENCODE BAM files used in this tutorial are already pre-processed (see <a href="../08_bam_quality_metrics/">Section 08</a>). They have been deduplicated and filtered for mapping quality (MAPQ â‰¥ 30), meaning multimappers have been removed.</p>
</blockquote>
<p>All outputs will be saved to the <code>deeptools_qc/</code> directory for organization.</p>
<h3 id="21-fingerprint-plot">2.1 Fingerprint Plot<a class="headerlink" href="#21-fingerprint-plot" title="Permanent link">&para;</a></h3>
<p><strong>Requirements:</strong> BAM files in <code>encode_bam/</code> directory.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Create output directory</span>
mkdir<span class="w"> </span>-p<span class="w"> </span>deeptools_qc

plotFingerprint<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-b<span class="w"> </span>encode_bam/ceb_ENCFF327JFG.bam<span class="w"> </span>encode_bam/H3K9ac_ENCFF193NPE.bam<span class="w"> </span>encode_bam/Input_ENCFF110SOB.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--skipZeros<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--numberOfSamples<span class="w"> </span><span class="m">50000</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-T<span class="w"> </span><span class="s2">&quot;Fingerprints of all BAM samples&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--plotFile<span class="w"> </span>deeptools_qc/fingerprints.pdf<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--plotFileFormat<span class="w"> </span>pdf<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--dpi<span class="w"> </span><span class="m">600</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>tee<span class="w"> </span>deeptools_qc/plotFingerprint.log
</code></pre></div>
<p><strong>What this does:</strong> Samples 50,000 random genomic positions and plots cumulative read distribution to assess IP enrichment. ChIP samples should show strong enrichment (steep curve), while Input should follow the diagonal.</p>
<h3 id="22-coverage-plot">2.2 Coverage Plot<a class="headerlink" href="#22-coverage-plot" title="Permanent link">&para;</a></h3>
<p>Checks sequencing depth and duplication levels.</p>
<div class="highlight"><pre><span></span><code>plotCoverage<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-b<span class="w"> </span>encode_bam/ceb_ENCFF327JFG.bam<span class="w"> </span>encode_bam/H3K9ac_ENCFF193NPE.bam<span class="w"> </span>encode_bam/Input_ENCFF110SOB.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-o<span class="w"> </span>deeptools_qc/coverage_histogram.pdf<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--plotFileFormat<span class="w"> </span>pdf<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--dpi<span class="w"> </span><span class="m">600</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--smartLabels<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--numberOfSamples<span class="w"> </span><span class="m">1000000</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--ignoreDuplicates<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--minMappingQuality<span class="w"> </span><span class="m">30</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--outRawCounts<span class="w"> </span>deeptools_qc/coverage_counts.txt
</code></pre></div>
<p><strong>What this does:</strong> Samples 1 million genomic positions and plots coverage distribution. The <code>--ignoreDuplicates</code> and <code>--minMappingQuality 30</code> flags document expected data quality (ENCODE BAMs are already filtered, so these serve as safety checks).</p>
<h3 id="23-summary-matrix">2.3 Summary Matrix<a class="headerlink" href="#23-summary-matrix" title="Permanent link">&para;</a></h3>
<p>We need a "Count Matrix" to compare samples. This counts reads in bins across the whole genome.</p>
<div class="highlight"><pre><span></span><code>multiBamSummary<span class="w"> </span>bins<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-b<span class="w"> </span>encode_bam/ceb_ENCFF327JFG.bam<span class="w"> </span>encode_bam/ceb_ENCFF744SVA.bam<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>encode_bam/H3K27me3_ENCFF164ALR.bam<span class="w"> </span>encode_bam/H3K27me3_ENCFF532DQH.bam<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>encode_bam/H3K9ac_ENCFF193NPE.bam<span class="w"> </span>encode_bam/H3K9ac_ENCFF534IPX.bam<span class="w"> </span><span class="se">\ </span><span class="w"> </span><span class="c1"># Input BAM files (IP samples only)</span>
<span class="w">  </span>--numberOfProcessors<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\ </span><span class="w">                      </span><span class="c1"># Run on 4 CPUs to speed up processing</span>
<span class="w">  </span>-o<span class="w"> </span>deeptools_qc/matrix.npz<span class="w"> </span><span class="se">\ </span><span class="w">                  </span><span class="c1"># Output compressed matrix file (for plotting)</span>
<span class="w">  </span>--outRawCounts<span class="w"> </span>deeptools_qc/matrix.tab<span class="w">         </span><span class="c1"># Output tab-separated counts file (readable text)</span>
</code></pre></div>
<p><strong>What this does:</strong> Divides the genome into bins and counts reads per bin for each sample, creating a matrix used by correlation and PCA analysis. We exclude Input samples to focus on comparing ChIP enrichment patterns.</p>
<h3 id="24-correlation-pca">2.4 Correlation &amp; PCA<a class="headerlink" href="#24-correlation-pca" title="Permanent link">&para;</a></h3>
<p>Using the matrix from Step 2.3, we compare the samples.</p>
<p><strong>Correlation Heatmap:</strong></p>
<div class="highlight"><pre><span></span><code>plotCorrelation<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-in<span class="w"> </span>deeptools_qc/matrix.npz<span class="w"> </span><span class="se">\ </span><span class="w">                 </span><span class="c1"># Input matrix from multiBamSummary</span>
<span class="w">  </span>--corMethod<span class="w"> </span>spearman<span class="w"> </span><span class="se">\ </span><span class="w">                        </span><span class="c1"># Use Spearman correlation (robust to outliers)</span>
<span class="w">  </span>--skipZeros<span class="w"> </span><span class="se">\ </span><span class="w">                                 </span><span class="c1"># Ignore bins with zero reads</span>
<span class="w">  </span>--whatToPlot<span class="w"> </span>heatmap<span class="w"> </span><span class="se">\ </span><span class="w">                        </span><span class="c1"># Generate a heatmap (vs scatterplot)</span>
<span class="w">  </span>--colorMap<span class="w"> </span>RdYlBu<span class="w"> </span><span class="se">\ </span><span class="w">                           </span><span class="c1"># Color map (Red-Yellow-Blue)</span>
<span class="w">  </span>--plotNumbers<span class="w"> </span><span class="se">\ </span><span class="w">                               </span><span class="c1"># Show correlation coefficients on plot</span>
<span class="w">  </span>--plotTitle<span class="w"> </span><span class="s2">&quot;Spearman correlation of binned genome coverage&quot;</span><span class="w"> </span><span class="se">\ </span><span class="c1"># Title</span>
<span class="w">  </span>--dpi<span class="w"> </span><span class="m">600</span><span class="w"> </span><span class="se">\ </span><span class="w">                                   </span><span class="c1"># High resolution for publication quality</span>
<span class="w">  </span>-o<span class="w"> </span>deeptools_qc/spearman_corr_plot.pdf<span class="w"> </span><span class="se">\ </span><span class="w">      </span><span class="c1"># Output PDF file</span>
<span class="w">  </span>--outFileCorMatrix<span class="w"> </span>deeptools_qc/spearman_corr_plot.tab<span class="w">  </span><span class="c1"># Save correlation values to text file</span>
</code></pre></div>
<p><strong>What this does:</strong> Computes pairwise Spearman correlation between samples using the count matrix. Biological replicates should show high correlation, and different marks/TFs should cluster separately.</p>
<p><strong>PCA (Principal Component Analysis):</strong></p>
<div class="highlight"><pre><span></span><code>plotPCA<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-in<span class="w"> </span>deeptools_qc/matrix.npz<span class="w"> </span><span class="se">\ </span><span class="w">                 </span><span class="c1"># Input matrix from multiBamSummary</span>
<span class="w">  </span>-o<span class="w"> </span>deeptools_qc/pca.pdf<span class="w"> </span><span class="se">\ </span><span class="w">                     </span><span class="c1"># Output PDF file</span>
<span class="w">  </span>-T<span class="w"> </span><span class="s2">&quot;PCA of samples based on binned genome coverage&quot;</span><span class="w"> </span><span class="se">\ </span><span class="c1"># Title</span>
<span class="w">  </span>--transpose<span class="w"> </span><span class="se">\ </span><span class="w">                                 </span><span class="c1"># Transpose matrix (required for correct grouping)</span>
<span class="w">  </span>--plotWidth<span class="w"> </span><span class="m">14</span><span class="w"> </span><span class="se">\ </span><span class="w">                              </span><span class="c1"># Width of the plot in inches</span>
<span class="w">  </span>--plotHeight<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="se">\ </span><span class="w">                             </span><span class="c1"># Height of the plot in inches</span>
<span class="w">  </span>--plotFileFormat<span class="w"> </span>pdf<span class="w"> </span><span class="se">\ </span><span class="w">                        </span><span class="c1"># File format</span>
<span class="w">  </span>--dpi<span class="w"> </span><span class="m">600</span><span class="w"> </span><span class="se">\ </span><span class="w">                                   </span><span class="c1"># High resolution for publication quality</span>
<span class="w">  </span>--outFileNameData<span class="w"> </span>deeptools_qc/pca.tab<span class="w"> </span><span class="se">\ </span><span class="w">      </span><span class="c1"># Save PCA coordinates to text file</span>
<span class="w">  </span>--markers<span class="w"> </span><span class="s1">&#39;s&#39;</span><span class="w"> </span><span class="s1">&#39;s&#39;</span><span class="w"> </span><span class="s1">&#39;D&#39;</span><span class="w"> </span><span class="s1">&#39;D&#39;</span><span class="w"> </span><span class="s1">&#39;o&#39;</span><span class="w"> </span><span class="s1">&#39;o&#39;</span><span class="w"> </span><span class="se">\ </span><span class="w">           </span><span class="c1"># Markers for each sample (pairs share markers)</span>
<span class="w">  </span>--colors<span class="w">  </span><span class="s1">&#39;#1b9e77&#39;</span><span class="w"> </span><span class="s1">&#39;#66c2a5&#39;</span><span class="w"> </span><span class="se">\ </span><span class="w">               </span><span class="c1"># Colors for ceb replicates (Green)</span>
<span class="w">            </span><span class="s1">&#39;#d95f02&#39;</span><span class="w"> </span><span class="s1">&#39;#fc8d62&#39;</span><span class="w"> </span><span class="se">\ </span><span class="w">               </span><span class="c1"># Colors for H3K27me3 replicates (Orange)</span>
<span class="w">            </span><span class="s1">&#39;#7570b3&#39;</span><span class="w"> </span><span class="s1">&#39;#8da0cb&#39;</span><span class="w">                  </span><span class="c1"># Colors for H3K9ac replicates (Purple)</span>
</code></pre></div>
<p><strong>What this does:</strong> Reduces high-dimensional count data to 2D for visualization. Samples should cluster by biological condition, with replicates grouping together and distinct marks/TFs separating along principal components.</p>
<p><strong>Using <code>sample_id.txt</code> for dynamic BAM file lists:</strong></p>
<p><strong>Requirements:</strong> You need <code>sample_id.txt</code> file containing sample IDs (one per line, created in section 08).</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Create BAM file paths from sample IDs</span>
<span class="nv">BAM_FILES</span><span class="o">=</span><span class="k">$(while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>sample<span class="p">;</span><span class="w"> </span><span class="k">do</span><span class="w"> </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;encode_bam/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.bam&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">done</span><span class="w"> </span>&lt;<span class="w"> </span>sample_id.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>tr<span class="w"> </span><span class="s1">&#39;\n&#39;</span><span class="w"> </span><span class="s1">&#39; &#39;</span><span class="k">)</span>

<span class="c1"># Create labels from basenames (for cleaner plot labels)</span>
<span class="nv">LABELS</span><span class="o">=</span><span class="k">$(</span>cat<span class="w"> </span>sample_id.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>tr<span class="w"> </span><span class="s1">&#39;\n&#39;</span><span class="w"> </span><span class="s1">&#39; &#39;</span><span class="k">)</span>

<span class="c1"># For IP samples only (excludes Input controls)</span>
<span class="nv">IP_FILES</span><span class="o">=</span><span class="k">$(</span>grep<span class="w"> </span>-v<span class="w"> </span><span class="s2">&quot;Input&quot;</span><span class="w"> </span>sample_id.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="k">while</span><span class="w"> </span><span class="nb">read</span><span class="w"> </span>sample<span class="p">;</span><span class="w"> </span><span class="k">do</span><span class="w"> </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;encode_bam/</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span><span class="s2">.bam&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">done</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>tr<span class="w"> </span><span class="s1">&#39;\n&#39;</span><span class="w"> </span><span class="s1">&#39; &#39;</span><span class="k">)</span>
<span class="nv">IP_LABELS</span><span class="o">=</span><span class="k">$(</span>grep<span class="w"> </span>-v<span class="w"> </span><span class="s2">&quot;Input&quot;</span><span class="w"> </span>sample_id.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>tr<span class="w"> </span><span class="s1">&#39;\n&#39;</span><span class="w"> </span><span class="s1">&#39; &#39;</span><span class="k">)</span>

<span class="c1"># Echo commands to verify what we built</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;All BAM files:&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="nv">$BAM_FILES</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;All labels:&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="nv">$LABELS</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;&quot;</span>
</code></pre></div>
<p><strong>What this script does (line-by-line):</strong></p>
<p><strong>For all samples (<code>BAM_FILES</code> and <code>LABELS</code>):</strong></p>
<ol>
<li>
<p><strong><code>BAM_FILES=...</code></strong> - Reads <code>sample_id.txt</code>, loops through each sample ID, prepends <code>encode_bam/</code> and appends <code>.bam</code>, then converts newlines to spaces. Result: space-separated list of full BAM file paths.</p>
</li>
<li>
<p><strong><code>LABELS=...</code></strong> - Reads <code>sample_id.txt</code> and converts newlines to spaces. Result: space-separated list of sample IDs (basenames) for cleaner plot labels.</p>
</li>
</ol>
<p><strong>For IP samples only (<code>IP_FILES</code> and <code>IP_LABELS</code>):</strong></p>
<p>The same logic applies, but <code>grep -v "Input"</code> filters out Input control samples first, creating IP-only file and label lists for correlation/PCA analysis.</p>
<p><strong>Echo demonstrations for <code>BAM_FILES</code> and <code>LABELS</code>:</strong></p>
<div class="highlight"><pre><span></span><code><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;All BAM files:&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="nv">$BAM_FILES</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;All labels:&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="nv">$LABELS</span>
</code></pre></div>
<p><strong>Expected output:</strong></p>
<div class="highlight"><pre><span></span><code>All BAM files:

encode_bam/ceb_ENCFF327JFG.bam encode_bam/ceb_ENCFF744SVA.bam encode_bam/H3K27me3_ENCFF164ALR.bam encode_bam/H3K27me3_ENCFF532DQH.bam encode_bam/H3K9ac_ENCFF193NPE.bam encode_bam/H3K9ac_ENCFF534IPX.bam encode_bam/Input_ENCFF110SOB.bam encode_bam/Input_ENCFF919XCV.bam

All labels:
ceb_ENCFF327JFG ceb_ENCFF744SVA H3K27me3_ENCFF164ALR H3K27me3_ENCFF532DQH H3K9ac_ENCFF193NPE H3K9ac_ENCFF534IPX Input_ENCFF110SOB Input_ENCFF919XCV
</code></pre></div>
<p>The basenames (labels) produce cleaner plot legends compared to full file paths.</p>
<p><strong>Example usage with labels:</strong></p>
<div class="highlight"><pre><span></span><code><span class="c1"># Use with plotFingerprint</span>
plotFingerprint<span class="w"> </span>-b<span class="w"> </span><span class="nv">$BAM_FILES</span><span class="w"> </span>--labels<span class="w"> </span><span class="nv">$LABELS</span><span class="w"> </span>--skipZeros<span class="w"> </span>...

<span class="c1"># Use with multiBamSummary (IP samples only)</span>
multiBamSummary<span class="w"> </span>bins<span class="w"> </span>-b<span class="w"> </span><span class="nv">$IP_FILES</span><span class="w"> </span>--labels<span class="w"> </span><span class="nv">$IP_LABELS</span><span class="w"> </span>...
</code></pre></div>
<hr />
<h2 id="level-3-reading-the-charts">Level 3: Reading the Charts<a class="headerlink" href="#level-3-reading-the-charts" title="Permanent link">&para;</a></h2>
<h3 id="31-interpreting-the-fingerprint-the-census"><a href="#21-fingerprint-plot">3.1 Interpreting the Fingerprint (The Census)</a><a class="headerlink" href="#31-interpreting-the-fingerprint-the-census" title="Permanent link">&para;</a></h3>
<p>This plot shows the cumulative read distribution across the genome. Good ChIP libraries show a clear separation between ChIP and input samples, with ChIP curves rising earlier due to enriched regions. Flat, overlapping curves usually indicate poor enrichment or over-background signal.</p>
<hr />
<p><img alt="Screenshot 2025-12-10 at 12 02 40â€¯PM" src="../images/fingerprint_plot.png" /></p>
<hr />
<table>
<thead>
<tr>
<th style="text-align: left;">Sample Type</th>
<th style="text-align: left;">Interpretation</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;"><strong>Input</strong></td>
<td style="text-align: left;">Close to the diagonal. Reads are uniformly distributed, behaving like ideal background.</td>
</tr>
<tr>
<td style="text-align: left;"><strong>H3K9ac</strong></td>
<td style="text-align: left;">Strong "Elbow". Reads concentrated in a small fraction of bins (focal peaks), showing a strong bend away from the diagonal.</td>
</tr>
</tbody>
</table>
<h3 id="32-interpreting-coverage"><a href="#22-coverage-plot">3.2 Interpreting Coverage</a><a class="headerlink" href="#32-interpreting-coverage" title="Permanent link">&para;</a></h3>
<p>Next, we look at the overall coverage distribution in each BAM using plotCoverage. This reveals whether some samples are globally under-sequenced, dominated by a few high-coverage regions, or heavily affected by duplicated reads. We restrict to high-quality, non-duplicate reads to make the distributions comparable.</p>
<p><strong>Plot A: The Drop-off</strong></p>
<ul>
<li>
<p><strong>Inputs :</strong> The Input tracks sit higher at low coverage because they spread their reads across the genome without enrichment. Thatâ€™s why both Input samples show a large fraction of bases at coverage 0 and 1, then taper off more slowly as coverage increases.</p>
</li>
<li>
<p><strong>ChIPs :</strong> In contrast, every IP sample collapses more sharply; the curves drop faster after coverage 1 because most genomic positions in a ChIP experiment receive almost no reads. Only a small portion of the genome â€” the actual binding or modification sites â€” reaches deeper coverage, and that fraction is tiny enough that the tail beyond coverage 2 nearly vanishes.</p>
</li>
</ul>
<hr />
<p><img alt="Screenshot 2025-12-10 at 12 03 51â€¯PM" src="../images/coverage_histogram_full.png" /></p>
<hr />
<p><strong>Plot B: The Tail</strong>
Zooming in reveals the difference. Input covers more of the genome at 1x depth, while ChIP focuses on peaks. The Input curves decline more slowly because a larger fraction of their genome maintains at least some measurable coverage. The IP curves fall off earlier and more steeply, which reflects the enrichment pattern: most positions have essentially no reads, and only a very small subset of bases in true peak regions sustain higher coverage.</p>
<hr />
<p><img alt="Screenshot 2025-12-10 at 12 04 14â€¯PM" src="../images/coverage_histogram_zoomed.png" /></p>
<hr />
<h3 id="33-interpreting-correlation-the-family-tree"><a href="#24-correlation--pca">3.3 Interpreting Correlation (The Family Tree)</a><a class="headerlink" href="#33-interpreting-correlation-the-family-tree" title="Permanent link">&para;</a></h3>
<p>Using the binned count matrix, we compute pairwise correlations between samples. A Spearman correlation heatmap shows whether biological replicates cluster together and whether inputs are distinct from ChIP samples. Poor clustering or scattered correlations usually indicate sample swaps, failed IPs, or inconsistent library prep.</p>
<p><strong>The Heatmap:</strong></p>
<ul>
<li><strong>Clustering:</strong> H3K27me3 samples cluster together and show moderately high mutual correlations (around 0.6), which is exactly what you expect for a broad repressive mark. The H3K9ac samples also correlate strongly with each other (0.76â€“1.0), forming a clean sub-cluster that is distinct from H3K27me3.</li>
<li><strong>Separation:</strong> Active marks (H3K9ac) should look different from Repressive marks (H3K27me3).</li>
</ul>
<hr />
<p><img alt="Screenshot 2025-12-10 at 12 05 28â€¯PM" src="../images/correlation_heatmap.png" /></p>
<hr />
<h3 id="34-interpreting-pca"><a href="#24-correlation--pca">3.4 Interpreting PCA</a><a class="headerlink" href="#34-interpreting-pca" title="Permanent link">&para;</a></h3>
<p>Finally, we perform PCA on the same binned count matrix. PCA reduces the data to a few dimensions that capture most of the variance. In a good ChIP-seq dataset, biological replicates cluster together in PCA space, and distinct conditions or marks separate along major components. PCA is a convenient visual check for batch effects, sample swaps, and outlier libraries.</p>
<p><strong>The Map:</strong></p>
<ul>
<li><strong>PC1 (X-axis):</strong> The PCA shows clear separation of samples by assay type. PC1 captures most of the variance and cleanly splits the H3K27me3 group from the H3K9ac group, which is expected because these marks have very different genomic distributions.</li>
<li><strong>Clustering:</strong> The two â€œcebâ€ samples cluster tightly together, indicating consistent coverage patterns within that group. The H3K27me3 replicates are also tightly paired, which matches their broad and uniform enrichment profile. The H3K9ac replicates sit on the opposite side of PC1, with one replicate shifted slightly on PC2, hinting at a mild difference in coverage distribution but nothing severe. If one replicate is far away, it might be an outlier/bad sample.</li>
</ul>
<hr />
<p><img alt="Screenshot 2025-12-10 at 12 05 56â€¯PM" src="../images/pca_plot.png" /></p>
<hr />
<h2 id="directory-structure-after-deeptools-qc">Directory Structure After deepTools QC<a class="headerlink" href="#directory-structure-after-deeptools-qc" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>chipseq_tutorial/
â”œâ”€â”€ encode_bam/                  â† ENCODE BAM files
â”‚   â”œâ”€â”€ ceb_ENCFF327JFG.bam
â”‚   â”œâ”€â”€ H3K9ac_ENCFF193NPE.bam
â”‚   â””â”€â”€ ...                      (8 BAM files total)
â”œâ”€â”€ encode_bam_qc/               â† QC from section 08
â”‚   â””â”€â”€ ...
â”œâ”€â”€ spp_qc/                      â† QC from section 09
â”‚   â””â”€â”€ ...
â”œâ”€â”€ deeptools_qc/                â† deepTools outputs
â”‚   â”œâ”€â”€ fingerprints.pdf
â”‚   â”œâ”€â”€ plotFingerprint.log
â”‚   â”œâ”€â”€ coverage_histogram.pdf
â”‚   â”œâ”€â”€ coverage_counts.txt
â”‚   â”œâ”€â”€ matrix.npz
â”‚   â”œâ”€â”€ matrix.tab
â”‚   â”œâ”€â”€ spearman_corr_plot.pdf
â”‚   â”œâ”€â”€ spearman_corr_plot.tab
â”‚   â”œâ”€â”€ pca.pdf
â”‚   â””â”€â”€ pca.tab
â””â”€â”€ sample_id.txt
</code></pre></div>
<hr />
<h2 id="summary">Summary<a class="headerlink" href="#summary" title="Permanent link">&para;</a></h2>
<ol>
<li><strong>Fingerprint:</strong> Confirms your IP worked (Sharp elbow).</li>
<li><strong>Coverage:</strong> Confirms sequencing depth (Inputs = broad, ChIP = peaky).</li>
<li><strong>PCA/Correlation:</strong> Confirms your replicates match (Siblings cluster together).</li>
</ol>
<blockquote>
<p>[!NOTE]
<strong>Up Next:</strong> With comprehensive QC validation complete, we're ready to call peaks with MACS2 and identify protein-DNA bindin sites.</p>
</blockquote>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 Raja Ishaq Nabi Khan
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/ishaaq34" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.top", "search.suggest", "search.highlight", "content.code.copy"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>